<!doctype html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <title>Deconvolution of Spatial Transcriptomics Using scRNA-seq</title>

    <meta name="description" content="Deconvolution of Spatial Transcriptomics using scRNA-seq: RCTD and Cell2location">
    <meta name="author" content="Wenxin Jiang">

    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <link rel="stylesheet" href="dist/reset.css">
    <link rel="stylesheet" href="dist/reveal.css">
    <link rel="stylesheet" href="dist/theme/league.css" id="theme">
    <!-- <link rel="stylesheet" href="dist/theme/black.css" id="theme"> -->
    <link rel="stylesheet" href="css/custom.css">
</head>

<body>
    <div class="reveal">
        <div class="slides">

            <section>
                <h2>Deconvolution of Spatial Transcriptomics Using scRNA-seq</h2>
                <p style="text-align: center;">Wenxin Jiang</p>
                <p style="text-align: center;">Oct. 18, 2024</p>
                <aside class="notes">
                    Note write here.
                </aside>
            </section>

            <section>
                <h2>How to Use This Webpage</h2>

                <h3>Download the slides as a PDF via <a
                        href="https://lucajiang.github.io/ST_Deconv/?print-pdf">here</a>.</h3>
                <h3>Press <span style="color:plum;"><strong>?</strong></span> on your keyboard to view
                    shortcuts.</h3>
                <!-- <h3>Press <span style="color: paleturquoise;">esc</span> on your keyboard to enter navigation mode.</h3> -->
                <h3>
                    Press <span style="color: gold;">
                        <span class="fa fa-arrow-up"></span>
                        <span class="fa fa-arrow-down"></span>
                        <span class="fa fa-arrow-left"></span>
                        <span class="fa fa-arrow-right"></span></span>
                    on your keyboard to navigate.
                </h3>
                <h3>Click the <span style="color:#13DAEC;">&#9776;</span> button at the lower left to open the menu.
                </h3>
                <!-- <aside class="notes">
                    Instructions on interacting with the presentation interface.
                </aside> -->
            </section>

            <section>
                <h2>Structure of the Presentation</h2>
                <ol>
                    <li>Introduction to <a href="#/3">spatial
                            transcriptomics</a> and <a href="#/4">scRNA-seq</a></li>
                    <li><a href="#/6">Deconvolution methods</a>: <a href="#/7">RCTD</a> and <a
                            href="#/8">Cell2Location</a></li>
                    <li><a href="#/9">Comparison</a>: <a href="#/9/4">Simulation</a> and <a href="#/9/5">Real data
                            analysis</a>
                    </li>
                </ol>
                <aside class="notes">
                    First, we will provide an introduction to st and sc. Next, we will discuss deconvolution methods,
                    focusing on two key approaches: RCTD and
                    c2l. Finally, we will draw comparisons between these methods through both simulation
                    studies and real data analyses.
                </aside>
            </section>

            <section>
                <section data-background-image="img/FFPE_Hero_V2_RGB.webp">
                    <h2>Spatially Resolved Transcriptomics:</h2>
                    <h2>Method of the Year 2020</h2>
                    <!-- ! unknown error in print_pdf mode -->
                    <div style="position: fixed; bottom: 10px; width: 100%; text-align: center; font-size: 0.6em;">
                        Press the <span style="color: gold;">DOWN</span> arrow to continue
                    </div>
                    <aside class="notes">
                        In 2020, st was recognized as the 'Method of the Year' by Nature Methods. This
                        innovative technique bridges the gap between traditional histology and gwas analysis,
                        offering great insights into cellular organization and function.
                    </aside>
                </section>

                <section>
                    <h3>Comparison between Sequencing Methods</h3>
                    <div>
                        <img src="img/spsc_complex.png">
                        <p style="text-align: center;"><small>Understanding the complexity of tissue with bulk RNA-seq,
                                scRNA-seq, and spatial transcriptomics.</small></p>
                    </div>
                    <aside class="notes">
                        If we draw an analogy to a fruit tart to illustrate different sequencing methods.
                        Consider the entire tart:
                        extracting juice from the whole tart represents bulk RNA-seq, where we obtain an average
                        gene expression profile
                        without distinguishing individual components. Similarly, extracting juice from each type of
                        fruit is like ct specific bulk-seq, capturing expression from grouped cell types. Sc-RNA
                        sequencing can be compared to
                        examining individual pieces of fruit, as it allows us to analyze gene expression at the
                        sc level, revealing
                        heterogeneity within the tissue. St, on the other hand, not only provides
                        detailed cellular
                        composition but also maintains the spatial context, like to knowing the precise arrangement of
                        fruit pieces in the tart,
                        thus offering crucial spatial information in understanding tissue complexity.
                    </aside>
                </section>

                <section>
                    <h3>Spatial Transcriptomics Maps Gene Expression in Tissue</h3>
                    <div>
                        <img src="img/sp_gene2loc.png">
                        <!-- <p style="text-align: center;">Spatial transcriptomics = Spatial information + gene
                            expression</p> -->
                        <p style="text-align: center;"><small>Spatial transcriptomics combines the spatial information
                                of tissue sections with the gene expression data from RNA sequencing.</small></p>
                    </div>
                    <aside class="notes">
                        St integrates gene expression with spatial information. On the sequencing platform, each spot is
                        marked with unique
                        barcodes, linking the sequenced mRNA to precise locations within the tissue. This allows us to
                        accurately map gene expression data back onto the histological images, offering a clear visual
                        representation of where specific genes are active within the tissue architecture.
                    </aside>
                </section>

                <section>
                    <h3>Mapping of Cell Types in Spatial Transcriptomics</h3>
                    <div>
                        <img src="img/st_map.png" style="width: 65%;">
                        <p style="text-align: center;"><small>If there is a single cell type in a spot, we can directly
                                map the most similar cell type from the reference scRNA-seq data.</small></p>
                    </div>
                    <aside class="notes">
                        When a spot contains only one single cell type, mapping is straightforward because we can
                        directly
                        identify the most
                        similar cell type using reference scRNA-seq data or marker genes. However, when multiple cell
                        types are present in a
                        single spot—a common senario —the challenge is to determine which cell types
                        are there and in what
                        proportions. This is where deconvolution methods become essential.

                    </aside>
                </section>

                <section>
                    <h3>Example: Mouse Hippocampus in 10X Visium</h3>
                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 60%; text-align: center;">
                            <img src="img/sp_illu.png" style="width: 100%;">
                            <p><small>Each spot may encapsulate multiple cells or cell types.</small></p>
                        </div>
                        <div style="width: 50%; text-align: center;">
                            <img src="img/sp_data.png" style="width: 45%; margin-left: -10%;">
                            <p style="width: 100%; text-align: center;"><small>Gene expression matrix is <span
                                        style="color: gold;">sparse</span> count data.</small></p>
                        </div>
                    </div>
                    <p style="text-align: left;"><small>The size of the spots in Visium does not allow us to study gene
                            expression in individual cells. With mixed cell types in spots, we cannot directly map the
                            cell types to the spots.</small></p>
                    <aside class="notes">
                        Here is a typical 10x Visium dataset of the mouse hippocampus. Each spot may contain around 10
                        cells. The sequencing output is a sparse count matrix, where each row is a gene and each
                        column is a spot. Additionally, the spatial coordinates for each spot on the
                        tissue are known. Due to technical limitations, the sequencing depth is much lower than
                        that of bulk RNA-seq or scRNA-seq.
                    </aside>
                </section>

                <section>
                    <h3>Deconvolution</h3>
                    <h3>Mapping scRNA-seq to Spatial Transcriptomics</h3>
                    <div>
                        <img src="img/decov_illu.png">
                        <!-- <p style="text-align: center;"><small>Illustration of deconvolution process</small></p> -->
                    </div>
                    <p style="text-align: left;"><small>With the reference scRNA-seq data, we can estimate the cell type
                            composition in each spot of the spatial transcriptomics data.</small></p>
                    <aside class="notes">
                        Since we want to know the cell type composition in this spot, we need to use the reference
                        sc data to estimate the contributions of each cell type. This process is called
                        deconvolution.
                    </aside>
                </section>
            </section>

            <section>
                <section>
                    <h2>Single-Cell RNA Sequencing</h2>
                    <h3>(scRNA-seq)</h3>
                    <div>
                        <img src="img/sc_illu.png" style="width: 70%;">
                        <p style="text-align: center;"><small>ScRNA-seq data provides the gene expression profile of
                                individual cells, which is also sparse count data.</small></p>
                    </div>
                    <aside class="notes">
                        Sc provides a detailed gene expression profile for individual cells, allowing
                        for deeper understanding of cellular function and diversity. This data is a highly sparse count
                        matrix,
                        reflecting the high sequencing depth that is typically employed compared to spatial sequencing.
                        In this matrix,
                        each row is an individual cell, while each column corresponds to a specific gene. This setup can
                        be used to estimate the average gene expression for each cell type.
                    </aside>
                </section>

                <section>
                    <h3>Compare scRNA-seq with Spatial Transcriptomics</h3>
                    <ul>
                        <li><b>Spatial transcriptomics</b>:
                            <ul>
                                <li>Spatial information</li>
                                <li>Typically low cellular resolution</li>
                                <li>Sequence depth is limited</li>
                            </ul>
                        </li>
                        <li><b>ScRNA-seq</b>:
                            <ul>
                                <li>Single-cell resolution</li>
                                <li>High sequence depth</li>
                            </ul>
                        </li>
                    </ul>
                    <aside class="notes">
                        To sum up, st excels at providing spatial context but typically comes with
                        lower cellular
                        resolution and limited sequencing depth. In contrast, sc offers
                        single-cell resolution and
                        higher sequencing depth, allowing for more detailed insights into gene expression at the
                        cellular level. This comparison
                        highlights the trade-offs between the rich spatial information and the detailed
                        cellular profiles.
                    </aside>
                </section>
            </section>

            <section>
                <h2>Insights from Deconvolution Results</h2>
                <div>
                    <img src="img/sp_apply.png" style="width: 70%;">
                </div>
                <p><small>ScRNA-seq and spatial transcriptomics data increase our understanding of the roles of
                        specific cell subpopulations and their interactions in development, homeostasis, and
                        disease.</small></p>
                <aside class="notes">
                    Deconvolution enriches our understanding of cellular roles and interactions in key biological
                    processes. For instance,
                    it can reveal the spatial distribution of various cell types within healthy or diseased tissues,
                    illustrating changes in
                    gene expression during tissue development. Additionally, it provides insights into the spatial
                    architecture of tumors
                    and their interactions with immune cells, highlighting the dynamic relationships that influence
                    development,
                    homeostasis, and progress of diseases.
                </aside>
            </section>

            <section>
                <section>
                    <h2>Deconvolution Methods</h2>
                    <div>
                        <img src="img/deconv.png" style="width: 70%;">
                        <p style="text-align: center;"><small>Workflow of deconvolution. Up: discovery of cell types in
                                scRNA-seq data. Down: discovery of gene expression in spatial transcriptomics
                                data.</small>
                        </p>
                    </div>
                    <aside class="notes">
                        Then, I will introduce two deconvolution methods: RCTD and c2l. We begin by obtaining
                        the gene-cell type matrix from sc, alongside the gene-spot matrix derived from
                        st. The primary goal of these methods is to accurately estimate the cell type
                        composition present in each spot of the st data.
                    </aside>
                </section>

                <section>
                    <h3>Commonly Used Distribution Models</h3>
                    <div style="text-align: left; font-size: 0.8em;">
                        <ul>
                            <li><b>Characteristics of scRNA-seq data:</b> This data consists of count data with
                                over-dispersion and large ratio of zero.</li>
                            <li><b>Over-dispersion:</b> The variance is greater than the mean, which is typical in count
                                data.</li>
                            <li><b>Assumption for spatial data:</b> Gene expression in a spot is
                                considered the independent sum of gene expressions from different cell types.</li>
                            <li><b>Commonly utilized distribution models:</b> Negative Binomial (NB), Poisson, and
                                Zero-Inflated Negative
                                Binomial (ZINB).</li>
                            <li><b>Additive Property:</b> Poisson is inherently additive, while NB and ZINB are additive
                                when the
                                dispersion parameter is fixed.</li>
                        </ul>
                    </div>
                    <aside class="notes">
                        Given the specific characteristics of sc and spatial data, selecting appropriate
                        distribution models for analysis
                        is essential. First, sc data is defined by count data that often exhibits
                        over-dispersion, meaning the variance
                        is greater than the mean. For spatial data, we assume that the gene expression observed in a
                        spot represents an
                        independent sum of gene expressions from the various cell types present. The commonly used
                        distribution models
                        include Poi and NB, whose additive properties are crucial for
                        the deconvolution process
                    </aside>
                </section>

            </section>

            <section>
                <section>
                    <h2>RCTD: Robust Cell Type Deconvolution</h2>
                    <div>
                        <img src="img/rctdFig2a.png">
                        <p style="text-align: center;"><small>Illustration of the RCTD method</small></p>
                    </div>
                    <aside class="notes">
                        RCTD, which stands for Robust Cell Type Deconvolution, addresses the deconvolution challenge in
                        a straightforward and
                        effective manner.
                    </aside>
                </section>

                <section>
                    <h3>RCTD Notation of Indexing:</h3>
                    <ul>
                        <li>$s$: index of the spot in spatial dataset</li>
                        <li>$g$: index of the gene</li>
                        <li>$f$: index of the cell type</li>
                    </ul>
                    <aside class="notes">
                        Now, let’s explore the notation. To align with the upcoming c2l model, I’ve
                        adapted the original RCTD notation for this presentation, which is different from the
                        original paper.
                        We have s is the index of a spot, g is the index of a gene, and f is the index of a cell type.
                    </aside>
                </section>

                <section>
                    <h3>RCTD Notation of Dataset:</h3>
                    <b>Observed data:</b>
                    <ul>
                        <li>$d_{sg}$: observed gene expression count in spot $s$ for gene $g$</li>
                        <li>$y_s$: total transcript count in spot $s$, equals to $\sum_g d_{sg}$</li>
                        <li>$g_{fg}$: mean gene expression for gene $g$ in cell type $f$</li>
                    </ul>
                    <aside class="notes">
                        The observed data, or we can say input data, includes the gene expression count in spots d, the
                        total transcript count in spots y, and the mean gene expression in cell types g. g is estimated
                        from sc.
                    </aside>
                </section>

                <section>
                    <h3>RCTD Model</h3>
                    <p>
                        \[
                        \begin{aligned}
                        d_{sg}|\lambda_{sg} &\sim \text{Poisson}(y_s \lambda_{sg}) \\
                        \log(\lambda_{sg}) &= \alpha_{s} + \log(\sum_f w_{sf} g_{fg}) + m_{g} + \epsilon_{sg}
                        \end{aligned}
                        \]
                    </p>
                    <p style="text-align: left;">Goal: estimate $w_{sf}$ with $g_{fg}$ estimated from scRNA-seq data.
                    </p>
                    <aside class="notes">
                        RCTD models the observed gene expression count in spots d as a Poisson distribution
                        parameterized
                        by the total transcript count y and the expression level lambda. The logarithm of lambda is
                        defined by the spot-specific effect alpha, the proportion of cell type f in spot w, the mean
                        gene
                        expression for gene g in cell type f, the gene-specific effect m, and the error term epsilon.
                        The
                        primary objective is to estimate w using g obtained from scRNA-seq reference data.
                    </aside>
                </section>

                <section>
                    <h3>RCTD Notation of Dataset:</h3>
                    <b>Unknown parameters:</b>
                    <ul>
                        <li>$w_{sf}$: proportion of cell type $f$ in spot $s$. $\sum_f w_{sf} = 1$ and $w_{sf} \geq 0$
                        </li>
                        <li>$\lambda_{sg}$: parameter of the Poisson distribution for gene $g$ in spot $s$.</li>
                        <li>$\alpha_{s}$: fixed pixel-specific effect for spot $s$</li>
                        <li>$m_{g}$: gene-specific platform random effect for gene $g$</li>
                        <li>$\epsilon_{sg}$: residual error term for gene $g$ in spot $s$</li>
                    </ul>
                    <aside class="notes">
                        Since w is the proportion of cell type, the sum of w over cell type f should be 1 and it should
                        be non-negative.
                    </aside>
                </section>

                <section>
                    <h3>RCTD Prior Settings</h3>
                    <ul>
                        <li>$m_{g} \sim \text{Normal}(0, \sigma^2_{m})$</li>
                        <li>$\epsilon_{sg} \sim \text{Normal}(0, \sigma^2_{\epsilon})$.</li>
                    </ul>
                    <aside class="notes">
                        RCTD assumes the prior distribution for the gene-specific platform random effect m is normal
                        with
                        mean 0 and variance sigma squared m. The residual error term epsilon is also assumed to follow a
                        normal distribution with mean 0 and variance sigma squared epsilon.
                    </aside>
                </section>

                <section>
                    <h3>RCTD Model Fitting</h3>
                    <ol>
                        <li>
                            <p><b><span style="color:darkseagreen">Estimate mean gene expression</span></b> $g_{fg}$
                                from scRNA-seq data, denoted as $\hat{g}_{fg}$.
                            </p>
                        </li>
                        <li>
                            <p class="fragment"><b><span style="color:darkseagreen">Gene filtering</span></b>: Filter
                                out
                                uninformative genes based on $\hat{g}_{fg}$. Filter to about 3k genes to reduce
                                computational burden.</p>
                        </li>
                    </ol>
                </section>

                <section>
                    <h4>RCTD Model Fitting</h4>
                    <div style="font-size: 0.6em; text-align: left;">
                        3. <b><span style="color:darkseagreen">Platform effect normalization</span></b> Estimate $m_{g}$
                        for gene $g$ by summarizing the spatial transcriptomics as a single pseudo-bulk measurement $S_g
                        \equiv \sum_s d_{sg} \sim
                        \text{Poisson}(\bar{d}_{g})$ with
                        \[
                        \begin{aligned}
                        \log(\mathbb{E}[\bar{d}_{g}]|\lambda_{.g}) &= \log\left(\frac{1}{S} \sum_s y_s
                        \lambda_{sg}\right) \\
                        &= m_{g} + \log(\bar{y}\sum_f B_{fg} g_{fg})\\
                        &\approx m_{g} + \log(\bar{y}\sum_f w_{f} g_{fg}) + \log(w_0)
                        \end{aligned}
                        \]
                        where $B_{fg} = \frac{1}{S} \sum_s \frac{y_s}{\bar{y}} w_{sf}
                        \exp(\alpha_{s}+\epsilon_{sg})$
                        and $w_{f} = \frac{1}{S} \sum_s \frac{y_s}{\bar{y}} w_{sf}\alpha_{s}$.

                        Obtain the MLE of unknown parameters ($w_{f}, w_{0}, \sigma^2_{m}$) and solve for $m_{g}$.
                        Denote the predicted platform effect as $\hat{m}_{g}$.
                    </div>
                    <aside class="notes">
                        The most important and novel step is the platform effect normalization. We estimate the gene
                        specific platform random effect m for gene g by summarizing the st data as a
                        single pseudo-bulk measurement. With this approximation, we can obtain the platform effect
                        normalization term for each gene by MLE.
                    </aside>
                </section>

                <section>
                    <h4>RCTD Model Fitting</h4>
                    <ol start="4">
                        <li>
                            <p><b><span style="color:darkseagreen">Inference</span></b> Estimate
                                $w_{sf}, \alpha_{s}$ and $\sigma^2_{\epsilon}$ by MLE. Assume estimated $\hat{g}_{fg}$
                                and $\hat{m}_{g}$ are fixed.</p>
                        </li>
                        <li>
                            <p class="fragment"><b><span style="color:darkseagreen">Expected cell-type-specific gene
                                        expression</span></b> $\mathbb{E}[d_{sf}|w, d_{sg}]
                                =\frac{d_{sg}w_{sf}\hat{g}_{fg}}{\sum_{f'} w_{sf'}\hat{g}_{f'g}}$.
                            </p>
                        </li>
                    </ol>
                    <aside class="notes">
                        By assuming that the estimated parameters are fixed, we can infer the proportions of cell types
                        in each spot using the
                        MLE method. These proportions represent the primary output of the deconvolution process.
                        Additionally, we can further estimate the expected ct-specific gene expression, which
                        allows us to identify
                        Spatially Variable Genes associated with different cell types.
                    </aside>
                </section>

                <section>
                    <h3>Three Modes of RCTD Model</h3>
                    <div style="font-size: 0.8em;">
                        <ul>
                            <li>
                                <p><b><span style="color:darkseagreen">Doublet mode</span></b> Assigns
                                    1-2 cell types per spot and is recommended for technologies with high spatial
                                    resolution such as <b>Slide-seq</b> and <b>MERFISH</b>.
                                </p>
                            </li>
                            <li>
                                <p><b><span style="color:darkseagreen">Full mode</span></b> Assigns any
                                    number of cell types per spot and is recommended for technologies with poor spatial
                                    resolution such as <b>100-micron resolution Visium</b>.
                                </p>
                            </li>
                            <li>
                                <p><b><span style="color:darkseagreen">Multi mode</span></b> is an
                                    extension of doublet mode that can discover more than two cell types (up to a
                                    pre-specified amount) per spot as an alternative option to full mode.</p>
                            </li>
                        </ul>
                    </div>
                    <aside class="notes">
                        RCTD has three modes suitable for different spatial technologies. The doublet mode is
                        the most commonly used mode which assigns 1-2 cell types per spot. It also can be easily
                        extended to consider more cell types.
                    </aside>
                </section>

                <section>
                    <h3>Cell Type Identification by Model Selection</h3>
                    <p>Denote $\mathcal{L}(f)$ as the likelihood of the model with only $f$ th cell type and
                        $\mathcal{L}(f, f')$ as the likelihood of the model only with $f$th and $f'$th cell types. For
                        each spot $s$:
                    </p>
                    \[
                    \hat{f} = \arg\max_f \mathcal{L}(f) \quad \text{and} \quad \hat{f'} = \arg\max_{f'\neq \hat{f}}
                    \mathcal{L}(\hat{f}, f')
                    \]
                </section>

                <section>
                    <h3>Cell Type Identification by Model Selection</h3>
                    <div style="font-size: 0.8em;">
                        <p style="text-align: left;">Because we expect many pixels to be single cell types, we can
                            apply a <b>penalized approach similar to AIC</b> to decide between the two models. We select
                            the model maximizing:</p>
                        \[
                        \text{AIC}(\mathcal{M}) \equiv \mathcal{L}(\mathcal{M}) - V_p(\mathcal{M})
                        \]
                        <p style="text-align: left;">where $p$ represents the number of parameters (cell types) and $V$
                            represents the penalty weight. Select $V=25$ based on simulation studies.</p>
                    </div>
                </section>

                <section>
                    <h3>Confident and Unconfident Spots</h3>
                    <p>Condition: Existence of another pair of cell types $(f, f')$ (f can be equal to f') such that
                    </p>
                    \[|\mathcal{L}(\hat{f}, \hat{f'}) - \mathcal{L}(f, f')| < \delta\]</p>
                        <p style="text-align: left;">If the condition holds, the spot is <b>unconfident</b>,
                            else it is <b>confident</b>.
                        </p>
                        <!-- <p style="text-align: left;">
                            The threshold $\delta$ is set to 10 based on simulation studies.
                        </p> -->
                        <aside class="notes">
                            We can classify spots as confident or unconfident based on the difference in likelihoods
                            between
                            the two models. If the difference is less than a threshold $\delta$, the spot is considered
                            unconfident; otherwise, it is classified as confident.
                        </aside>
                </section>

                <section>
                    <h3>Sequential Quadratic Programming for MLE</h3>
                    <p>Aim: Convert nonlinear problems into a series of quadratic programming
                        (QP) problems.</p>
                    <div style="font-size: 0.8em;">
                        <p style="text-align: left;">The parameter $\alpha_s$ effectively allows us to rescale $w_s$,
                            so we define $w_{f, s}=w_{f, s} e^{\alpha_s}$, which <b>will not be constrained to sum to
                                1</b>. Next, define:</p>
                        \[
                        \bar{\lambda}_{s, g}\left(w_s\right)=y_s \sum_{f=1}^F w_{f, s} g_{f, g} e^{\hat{m}_g}=y_s
                        \sum_{f=1}^F
                        w_{f, s} \bar{g}_{f, g}
                        \]
                        <p style="text-align: left;">We will refer to this as the predicted mean of gene $g$ in pixel
                            $s$.</p>
                    </div>
                    <aside class="notes">
                        Sequential Quadratic Programming is a method used to solve nonlinear optimization problems by
                        converting them into a series of quadratic programming problems. In the context of RCTD, we
                        introduce the parameter alpha to rescale w, allowing us to define wfs as wfs times e to the
                        power of alpha. This transformation ensures that w will not be constrained to sum to 1. We
                        also define the predicted mean of gene g in pixel s as the sum of w times the mean gene
                        expression times e to the power of m hat g.
                    </aside>
                </section>

                <section>
                    <h4>SQP for MLE (cont.)</h4>
                    <div style="font-size: 0.8em;">
                        <p style="text-align: left;">The final model is a Poisson log-normal mixture model:</p>
                        \[
                        d_{s, g} \mid \bar{\lambda}_{s, g} \sim \operatorname{Poisson}\left(e^{\varepsilon_{s, g}}
                        \bar{\lambda}_{s, g}\left(w_s\right)\right), \quad \varepsilon_{s, g} \sim
                        \operatorname{Normal}\left(0,
                        \sigma_{\varepsilon}^2\right)
                        \]
                        <p style="text-align: left;">We estimate $w_s^* \geq 0$ as the solution that maximizes the
                            <b>log-likelihood</b>
                            $\mathcal{L}\left(w_s\right)$:
                        </p>
                        \[
                        \max \mathcal{L}\left(w_s\right)=\sum_{g=1}^G \log P\left(d_{s, g} \mid \bar{\lambda}_{s,
                        g}\left(w_s\right)\right) \quad \text { subject to: } w \geq 0
                        \]
                    </div>
                    <aside class="notes">
                        The final model is a Poisson log-normal mixture model. We estimate w* as the
                        solution that maximizes the log-likelihood of the model, subject to the constraint that w is
                        non-negative.
                    </aside>
                </section>

                <section>
                    <h4>SQP for MLE (cont.)</h4>
                    <div style="font-size: 0.8em;">
                        <p style="text-align: left;">From now on, we consider a fixed spot $s$ and suppress the notation
                            of $s$. We will estimate $w^* \geq 0$ as that which maximizes the <b>log-likelihood</b>
                            $\mathcal{L}(w)$:
                        </p>
                        \[
                        \max _w \mathcal{L}(w)=\sum_{g=1}^G \log P\left(d_g \mid \lambda_g(w)\right)=\sum_{g=1}^G \log
                        Q_{d_g}\left(\lambda_g(w)\right)
                        \]
                    </div>
                    <aside class="notes">
                        We will now focus on a fixed spot s. We aim to estimate w* as the value that maximizes the
                        log-likelihood of the model.
                    </aside>
                </section>

                <section>
                    <h4>SQP for MLE (cont.)</h4>
                    <div style="font-size: 0.8em;">
                        <p style="text-align: left;">Our log likelihood is non-convex. To optimize it, we will apply
                            Sequential Quadratic
                            Programming,
                            an
                            iterative procedure that will be repeated until convergence. Let $w_0$ be the value of $w$
                            at a
                            given iteration, and let the gradient of $-\mathcal{L}$ be $b(w)$ and the Hessian of
                            $-\mathcal{L}$
                            be $A(w)$. Then, we can make the following quadratic Taylor approximation to $\mathcal{L}$:
                        </p>
                        \[
                        \begin{aligned}
                        -\mathcal{L}(w)
                        \approx&-\mathcal{L}\left(w_0\right)+b\left(w_0\right)^T\left(w-w_0\right)\\
                        &+\frac{1}{2}\left(w-w_0\right)^T A\left(w_0\right)\left(w-w_0\right)
                        \end{aligned}
                        \]
                    </div>
                    <aside class="notes">
                        The log likelihood is non-convex, so we will use Sequential Quadratic Programming to optimize
                        it. We use a quadratic Taylor approximation to the log likelihood to iteratively update the
                        value of w until convergence.
                    </aside>
                </section>

                <section>
                    <h3>SQP for MLE: Construct Positive Semi-Definite Hessian</h3>
                    <p style="font-size: 0.8em; text-align: left;">This QP will not be well-behaved if the Hessian
                        $A\left(w_0\right)$ is not positive semi-definite, which can occur due to the non-convexity of
                        $-\mathcal{L}(w)$. Specifically, suppose we have an eigen-decomposition of $H$ as:</p>
                    \[
                    H=V D V^T
                    \]
                    <p style="font-size: 0.8em; text-align: left;">Here, $D$ is a diagonal matrix of eigenvalues. We
                        obtain the positive semi-definite part of H by taking $D^{+}=\max (D, 0)$ and:</p>
                    \[
                    A=V D^{+} V^T
                    \]
                    <aside class="notes">
                        The Hessian must be positive semi-definite for the QP to be well-behaved. If the Hessian is not
                        positive semi-definite, we can obtain the positive semi-definite part of the Hessian by taking
                        the maximum of the eigenvalues and using this to construct a new positive semi-definite Hessian.
                    </aside>
                </section>

                <section>
                    <h3>Advantages of RCTD</h3>
                    <ul>
                        <li>Models platform effects by taking spatial transcriptomics as a pseudo-bulk
                            measurement.
                        </li>
                        <li>Robust to varying UMI counts per pixel, multiple cell types per pixel, and missing cell
                            types in
                            the reference.</li>
                    </ul>
                    <aside class="notes">
                        RCTD has several advantages, including its ability to model platform effects by summarizing
                        spatial transcriptomics as a pseudo-bulk measurement. It is also robust to varying UMI counts
                        per pixel, multiple cell types per pixel, and missing cell types in the reference.
                    </aside>
                </section>

                <section>
                    <h3>Platform Effect Estimation</h3>
                    <div style="text-align:center;">
                        <img src="img/rctdFig1f.png" alt="Figure 1f" style="width:40%;">
                        <p><small>Density plot across genes of measured platform effects between cerebellum
                                <b>scRNA-seq</b> and <b>snRNA-seq (single nucleus RNA-seq)</b> data.</small></p>
                    </div>
                    <aside class="notes">
                        Platform effect occurs when transferring cell type information from sc to st. Since we don't
                        have the ground truth of spatial data, we can compare sc and single nucleus to estimate the
                        platform effect. The x axis is the log ratio of the gene expression in sc and sn, and the y axis
                        is the density. If there is no platform effect, the density should be centered around 0.
                    </aside>
                </section>

                <section>
                    <h3>Platform Effect Estimation</h3>
                    <div style="text-align:center; font-size: 0.6em;">
                        <p>In the 3rd step of model fitting, RCTD estimates the platform effect of each gene to transfer
                            cell type information from scRNA-seq to spatial transcriptomics by taking the spatial
                            transcriptomics as a bulk measurement.
                        </p>
                    </div>

                    <div style="text-align:center;">
                        <img src="img/rctdFig2bc.png" alt="Figure 2bc" style="width:60%;">
                        <p><small>Left: Scatter plot of measured versus predicted platform effect for each gene between
                                the sc and sn cerebellum datasets. Right: Confusion matrix for RCTD’s performance on
                                cross-platform (trained on snRNA-seq data and tested on scRNA-seq data).</small></p>
                    </div>
                    <aside class="notes">
                        Here we compare the measured and predicted platform effects for each gene between sc and sn
                        cerebellum datasets. The left plot shows the scatter plot of measured versus predicted platform
                        effects. The right plot shows the confusion matrix for RCTD's performance on cross-platform.
                        This illustrates the effectiveness of RCTD in estimating the platform effect.
                    </aside>
                </section>

                <section>
                    <h3>Robustness</h3>
                    <div style="font-size: 0.8em;">
                        <ul>
                            <li>
                                <p><b><span style="color:darkseagreen">Varying unique molecular
                                            identifier (UMI) counts per pixel:</b></span> Additional UMIs per
                                    pixel lead to an increased confidence rate.</p>
                            </li>
                            <li>
                                <p class="fragment"><b><span style="color:darkseagreen">Multiple cell types per
                                            pixel:</b></span> RCTD was able to accurately predict cell class
                                    proportions on
                                    pixels containing three or four cell types.</p>
                            </li>
                            <li>
                                <p class="fragment"><b><span style="color:darkseagreen">Missing cell types in the
                                            reference:</b></span>When cell types in the simulated spatial data were
                                    missing
                                    from the reference, RCTD classified pixels as the most transcriptionally similar
                                    cell
                                    type in the reference if available. When no closest cell type was available in the
                                    reference, RCTD predicted cell types with reduced confidence rates but often
                                    misclassified such pixels.</p>
                            </li>
                        </ul>
                    </div>
                    <aside class="notes">
                        RCTD is robust to varying UMI counts per pixel, multiple cell types per pixel, and missing cell
                        types in the reference. It can accurately predict cell class proportions
                    </aside>
                </section>

                <!-- <section>
                    <h3>Application: Detecting Spatially Variable Genes and Effect of Cellular Environment</h3>
                    <p style="text-align: left; font-size: 0.8em;">Differentiate cell type marker genes from spatially
                        variable genes.</p>

                    <div style="text-align:center;">
                        <img src="img/rctdFig6d.png" alt="Figure 6d" style="width:70%;">
                        <p><small>Smoothed spatial expression patterns, recovered by local regression, of two genes
                                detected to have large spatial variation within RCTD’s CA3 cells.</small></p>
                    </div>
                    <aside class="notes">
                        With a robust deconvolution result, we can differentiate cell type marker genes from spatially variable genes. The plot shows the smoothed spatial expression patterns of two genes detected to have large spatial variation within RCTD's CA3 cells.
                    </aside>
                </section> -->

                <section>
                    <h3>Limitations</h3>
                    <ol>
                        <li>Assumes that platform effects are shared across cell types and random noise $\epsilon_{fg}$
                            is
                            shared across genes.</li>
                        <li>Hard to handle cases when the cell type is missing from the reference but exists in the
                            spatial
                            data.</li>
                    </ol>
                    <aside class="notes">
                        RCTD has some limitations. It assumes that platform effects are shared across cell types and
                        random noise is shared across genes. It also struggles to handle cases when the cell type is
                        missing from the reference but exists in the spatial data.
                    </aside>
                </section>
            </section>

            <section>
                <section>
                    <h2>Cell2Location</h2>
                    <div>
                        <img src="img/c2lFig1a.png">
                        <p style="text-align: center;"><small>Illustration of the Cell2Location method</small></p>
                    </div>
                    <aside class="notes">
                        To address the limitations of RCTD, c2l was developed with a flexible model that carefully
                        models different sources of variation in st data.
                    </aside>
                </section>

                <section>
                    <h3>Cell2Location Notation of Index</h3>
                    <ul>
                        <li>$g \in \{1, \ldots, G\}$: Gene</li>
                        <li>$s \in \{1, \ldots, S\}$: Spot</li>
                        <li>$e \in \{1, \ldots, E\}$: Spatial dataset</li>
                        <li>$f \in \{1, \ldots, F\}$: Cell type</li>
                    </ul>
                    <aside class="notes">
                        Still, g is the index of a gene, s is the index of a spot and f is the index of a cell type.
                        Different from RCTD, c2l can handle multiple spatial datasets. We use e to index the different
                        spatial experiments.
                    </aside>
                </section>

                <section>
                    <h3>Cell2Location Notation of Dataset</h3>
                    <p><b>Input</b>: Matrix of reference expression levels $g_{fg}$ and matrix of spatial expression
                        counts
                        $d_{sg}$.</p>
                    <p><b>Main output</b>: Cell abundance $w_{sf}$ which can be transformed into cell type
                        proportions.
                    </p>
                    <aside class="notes">
                        The input data includes the matrix of reference expression levels g and the matrix of spatial
                        expression counts d. The primary output is the cell abundance w, which can be transformed
                        into cell type proportions. Here, we do not assume that the sum of w over cell type f is 1,
                        that's why we call it cell abundance.
                    </aside>
                </section>

                <section>
                    <h3>Cell2Location Model of mRNA Counts</h3>
                    <div style="font-size: 0.8em;">
                        \[
                        d_{sg} \sim \text{NB}(\mu_{sg}, \alpha_{eg}), \quad
                        \mu_{sg} = \left( m_{g} \sum_{f} w_{sf} g_{fg} + s_{eg} \right) y_s
                        \]
                    </div>
                    <div style="text-align: center;">
                        <img src="img/c2lSFig1a.png" alt="SFigure 1f" style="width: 90%;">
                    </div>
                    <aside class="notes">
                        c2l model d as a negative binomial distribution with mean mu and overdispersion parameter alpha
                        which is flexible than Poisson distribution in RCTD. The mean mu is defined by the technology
                        sensitivity parameter m, the regression weights w, the reference expression levels g, the
                        gene-specific additive shift s, and the location-specific scaling factor y. The figure shows the
                        probabilistic graphical model. The most important part is that c2l factorizes the cell abundance
                        prior into two parts.
                    </aside>
                </section>

                <section>
                    <p><b>Cell2Location Notation of Model Parameters</b></p>
                    <div style="font-size: 0.9em;">
                        <ul>
                            <li>$d_{sg}$: Spatial expression of gene $g$ in spot $s$</li>
                            <li>$\mu_{sg}$: Unobserved expression level (rate) of gene $g$ in spot $s$</li>
                            <li>$\alpha_{eg}$: Gene- and batch-specific over-dispersion parameter</li>
                            <li>$m_{g}$: Technology sensitivity parameter for gene $g$</li>
                            <li>$w_{sf}$: Regression weight for cell type $f$ in spot $s$</li>
                            <li>$g_{fg}$: Reference expression level of gene $g$ in cell type $f$</li>
                            <li>$s_{eg}$: Gene-specific additive shift in spatial dataset $e$</li>
                            <li>$y_s$: Location-specific scaling factor for spot $s$</li>
                        </ul>
                    </div>
                    <aside class="notes">
                        SKIP.
                    </aside>
                </section>

                <section>
                    <h3>Cell2Location Cell Abundance Prior</h3>
                    <img src="img/c2lSFig1b.png" alt="SFigure 1f" style="width:90%;">
                    <aside class="notes">
                        c2l factorizes the cell abundance prior into x and z by assuming there are several colocated
                        ct latent group r for each ct. This concept can be visualized as a two-dimensional clustering
                        that
                        relates to both ct and spatial location.
                        Within each latent group, the ct are assumed to have similar spatial distributions,
                        while the spots within these groups are expected to exhibit similar ct compositions.
                    </aside>
                </section>

                <section>
                    <h3>Prior Settings of Cell2Location</h3>
                    <p style="font-size: 0.8em; text-align: left;">Gene-specific multiplicative scaling factor</p>
                    <div style="font-size: 0.6em;">
                        \[
                        \begin{aligned}
                        m_g &\sim \text{Gamma}\left(\alpha^m, \alpha^m/\mu^m \right)
                        \left\{ \begin{array}{ll}
                        \alpha^m & = 1/(o^m)^2, \quad o^m \sim \text{Exp}(3) \\
                        \mu^m &\sim \text{Gamma}(1, 1) \\
                        \end{array} \right. \\
                        \end{aligned}
                        \]
                    </div>

                    <p style="font-size: 0.8em; text-align: left;">The prior on detection efficiency per location</p>
                    <div style="font-size: 0.6em;">
                        \[
                        \begin{aligned}
                        y_s &\sim \text{Gamma}\left(\alpha^y, \alpha^y/y_e \right) \quad y_e \sim \text{Gamma}(10,
                        10/\mu^y)
                        \end{aligned}
                        \]
                    </div>

                    <p style="font-size: 0.8em; text-align: left;">Overdispersion for each gene</p>
                    <div style="font-size: 0.6em;">
                        \[
                        \alpha_{eg} = 1/o_{eg}^2, \quad o_{eg} \sim \text{Exp}(\beta^o), \quad \beta^o \sim
                        \text{Gamma}(9,
                        3)
                        \]
                    </div>
                    <aside class="notes">
                        Most of the priors are set as gamma distributions which is the conjugate prior for the
                        negative binomial distribution.
                    </aside>
                </section>

                <section>
                    <p style="font-size: 0.8em; text-align: left;">Additive shift for genes</p>
                    <div style="font-size: 0.6em;">
                        \[
                        \begin{aligned}
                        s_{eg} &\sim \text{Gamma}\left(\alpha^s_e, \alpha^s_e/\mu^s_e \right)
                        \left\{ \begin{array}{ll}
                        \mu^s_e &\sim \text{Gamma}(1, 100) \\
                        \alpha^s_e &=1/o_e^2, \quad o_e \sim \text{Exp}(\beta^s), \quad \beta^s \sim \text{Gamma}(9, 3)
                        \end{array} \right.
                        \end{aligned}
                        \]
                    </div>

                    <p style="font-size: 0.8em; text-align: left;">Cell abundance prior:</p>
                    <div style="font-size: 0.6em;">
                        \[
                        \begin{aligned}
                        w_{sf} &\sim \text{Gamma}\left(\sum_r z_{sr} x_{rf} \nu^w, \nu^w\right) \\
                        z_{sr} &\sim \text{Gamma}\left(B_s/R, 1/(N_s/B_s)\right)
                        \left\{ \begin{array}{ll}
                        N_s &\sim \text{Gamma}\left(\hat{N} \nu^n, \nu^n\right) \\
                        B_s &\sim \text{Gamma}\left(\hat{B}, 1\right) \\
                        \end{array} \right. \\
                        x_{rf} &\sim \text{Gamma}\left(K_r/R, K_r\right), \quad K_r \sim
                        \text{Gamma}\left(\hat{A}/\hat{B},
                        1\right) \\
                        \end{aligned}
                        \]
                    </div>
                    <aside class="notes">
                        SKIP
                    </aside>
                </section>

                <section>
                    <h3>Cell2Location Model: Probabilistic Graph</h3>
                    <div style="text-align:center;">
                        <img src="img/c2l_pm.png" alt="Probabilistic Graph" style="width:85%;">
                    </div>
                    <aside class="notes">
                        Combining the prior settings, we can construct the probabilistic graph of the c2l model.
                        The hyperparameters colored in grey have default values while the hyperparameters colored in
                        black are obtained from the data.
                    </aside>
                </section>

                <section>
                    <h3>Strength of the Cell2Location Model</h3>
                    <p style="font-size: 0.8em; text-align: left;">Joint modeling of multiple spatial
                        experiments / batches provides several benefits due to normalization and sharing of information
                        between experiments:</p>

                    <div style="font-size: 0.8em;">
                        <ul>
                            <li>
                                <p>Modeling differences in RNA detection sensitivity across
                                    experiments: $y_e$ and $y_s$.</p>
                            </li>
                            <li>
                                <p>Increasing accuracy by improving the model's ability to
                                    distinguish low sensitivity
                                    $m_g$ from zero cell abundance $w_{rf}$.</p>
                            </li>
                            <li>
                                <p>Increasing sensitivity by sharing factorization prior on cell
                                    abundance $w_{rf}$, namely
                                    which cell types tend to co-locate across experiments represented by $x_{rf}$.</p>
                            </li>
                        </ul>
                    </div>
                    <aside class="notes">
                        c2l are able to model multiple spatial experiments simultaneously, which provides more robust
                        estimation.
                    </aside>
                </section>

                <section>
                    <h3>Fitting the Cell2Location Model</h3>
                    <h4>Step 1: Construction of Reference Cell Type Signatures</h4>

                    <div style="font-size: 0.8em;">
                        <ul>
                            <li>
                                <p>
                                    <scan style="color:darkseagreen">Distribution</scan>: By default,
                                    the Cell2Location employs a <b>negative binomial regression</b> to estimate
                                    reference cell type signatures, which allows robust combination of data from
                                    multiple
                                    sources.
                                </p>
                            </li>
                            <li>
                                <p>
                                    <scan style="color:darkseagreen">Mean</scan>: Alternatively, a
                                    <b>hand-coded method</b> that estimates the average expression of each gene in each
                                    cell type can be used.
                                </p>
                            </li>
                        </ul>
                    </div>
                    <aside class="notes">
                        Similar to RCTD, the first step is to construct reference cell type signatures. However, c2l
                        provides an alternative method to model the distribution of reference ct as a negative binomial
                        distribution. This result in a realistic modeling of the data.
                    </aside>
                </section>

                <section>
                    <h4>Fitting the Cell2Location Model</h4>
                    <h4>Step 2: Determining the Hyperparameters</h4>
                    <div style="font-size: 0.8em;">
                        <ul>
                            <li>
                                <p>
                                    <scan style="color:darkseagreen">Expected cell abundance $\hat{N}$ per location
                                    </scan>: Can be derived from histology images or tissue type.
                                </p>
                            </li>
                            <li>
                                <p>
                                    <scan style="color:darkseagreen">Hyperparameter $\alpha^y$ for regularizing
                                        within-experiment variation in RNA detection sensitivity</scan>: If not strong
                                    within-experiment variability in RNA detection sensitivity across locations, set
                                    $\alpha^y=200$. Otherwise, set $\alpha^y=20$.
                                </p>
                            </li>
                        </ul>
                    </div>
                    <aside class="notes">
                        There are 2 hyperparameters that need to be determined before fitting the model. The expected
                        cell abundance per location can be derived from histology images or tissue type. The
                        hyperparameter alpha y is determined based on spatial platform characteristics.
                    </aside>
                </section>

                <section>
                    <h4>Fitting the Cell2Location Model</h4>
                    <h4>Step 3: Variational Inference</h4>
                    <p style="text-align: left; font-size: 0.8em;">Variational Bayesian Inference is used to approximate
                        the posterior, building on the Automatic Differentiation Variational Inference (<scan
                            style="color:darkseagreen">ADVI</scan>) framework
                        implemented in Pyro.</p>
                    <p style="text-align: left; font-size: 0.6em;">The posterior distribution over unknown parameters
                        is approximated by softplus-transformed (to ensure a positive scale) <scan
                            style="color:darkseagreen">univariate normal</scan> distributions.</p>
                    <p style="text-align: left; font-size: 0.6em;">
                        <scan style="color:darkseagreen">Minimizing the KL divergence</scan> between the variational
                        approximation and the true posterior
                        distribution (or maximizing ELBO).
                    </p>
                    <aside class="notes">
                        The prior settings of c2l is too complex, which makes it hard to derive the posterior
                        distribution with vi. Here we use ADVI frame work implemented in Pyro to approximate each
                        posterior as a univariate normal distribution.
                    </aside>
                </section>

                <section>
                    <h3>Brief Introduction to Pyro</h3>
                    <p style="text-align: left; font-size: 0.8em;">Here's an example to illustrate the use of Pyro for
                        inference:</p>

                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 73%; text-align: center;">
                            <div style="font-size: 0.7em;">
                                <pre><code data-line-numbers="5,6,8,11,13-16" class="hljs python"">data = np.random.binomial(1, 0.8, size=200) # random-generated data
                    
def model(data): # define the probabilistic model
    # define the hyperparameters that control the Beta prior
    alpha0 = pyro.param("alpha0", torch.tensor(10.0))
    beta0 = pyro.param("beta0", torch.tensor(10.0))
    # sample f from the Beta prior
    f = pyro.sample("latent_var", dist.Beta(alpha0, beta0))
    # sample observed data from the Bernoulli likelihood
    with pyro.plate("data", len(data)):
        pyro.sample("obs", dist.Bernoulli(f), obs=data)

svi = SVI(model,
            AutoDiagonalNormal(model),
            Adam({"lr": 0.0005, "betas": (0.90, 0.999)}),
            loss=Trace_ELBO())

[svi.step(data) for _ in range(1000)] # run inference
                                </code></pre>
                            </div>
                        </div>
                        <div style="width: 30%; text-align: right; margin-left: -8em;">
                            <img src="img/pyro_structure.svg" alt="Pyro structure" style="width: 100%;">
                        </div>
                    </div>
                    <aside class="notes">
                        Here is an example of how to use Pyro for inference. We first generate some random data and
                        define the probabilistic model. The latent variable is sampled from the Beta prior with
                        parameters
                        alpha0 and beta0. The observed data is sampled from the Bernoulli likelihood with the latent
                        variable as the parameter. We then use the SVI class to perform inference using the
                        AutoDiagonalNormal
                        guide which approximates the posterior as a univariate normal distribution. We use the Adam
                        optimizer
                        to minimize the ELBO loss function. The usage of Pyro makes it easy to define complex
                        probabilistic models.
                    </aside>
                </section>

                <section>
                    <h3>Features of Cell2Location</h3>
                    <div style="font-size: 0.8em;">
                        <ul>
                            <li>
                                <p><b><span style="color:darkseagreen">Borrow statistical strength
                                            across locations</span> with similar cell composition</b></p>
                            </li>
                            <li>
                                <p class="fragment">Account for <b><span style="color:darkseagreen">batch
                                            variation</span>
                                        across slides as well as variation in <span style="color: darkseagreen;">mRNA
                                            detection sensitivity<span></b>
                                </p>
                            </li>
                            <li>
                                <p class="fragment"><b>Estimate absolute cell type abundances by incorporating <span
                                            style="color:darkseagreen">prior information</span> about the analyzed
                                        tissues</b>
                                </p>
                            </li>
                            <li>
                                <p class="fragment"><b><span style="color:darkseagreen">Computationally
                                            efficient</span>,
                                        owing to variational approximate inference and GPU acceleration</b>
                                </p>
                            </li>
                            <li>
                                <p class="fragment"><b>Integrated with the
                                        <span style="color:darkseagreen">scvi-tools</span> framework and comes with a
                                        suite of downstream analysis
                                        tools</b></p>
                            </li>
                        </ul>
                    </div>
                    <aside class="notes">
                        To summarize, c2l has several advantages, including the ability to borrow statistical strength
                        across locations with similar cell composition, account for batch variation and mRNA detection
                        sensitivity, estimate absolute cell type abundances, computational efficiency, and integration
                        with the scvi-tools framework, providing access to a suite of downstream analysis tools.
                    </aside>
                </section>

                <section>
                    <h3>Sensitivity Analysis of Cell2Location</h3>
                    <p style="text-align: left; font-size: 0.8em;">Cell2Location is sensitive to the choice of
                        $\alpha^y$ and $\hat{N}$ but robust to $\hat{A}$ and
                        $\hat{B}$.</p>
                    <div style="text-align:center;">
                        <img src="img/c2lsena.png" alt="SFigure 5" style="width:90%;">
                        <!-- <img src="img/c2lsenb.png" alt="SFigure 5" style="width:90%;"> -->
                        <!-- <p><small><b>SupFigure 5</b>: Stability of cell2location cell abundance estimates with respect
                                to
                                model hyperparameter settings.</small></p> -->
                    </div>
                    <aside class="notes">
                        c2l is sensitive to the choice of alpha y and hat N but robust to hat A: the expected number of
                        cell types per location and hat B: the expected group size of locations.
                    </aside>
                </section>

                <section>
                    <h4>Ablation Study and Unaligned Cell Type Signatures</h4>
                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 70%; text-align: center;">
                            <img src="img/c2lablation.png" alt="SFigure 4" style="width:100%;">
                        </div>

                        <div style="width: 30%; text-align: left; margin-left:-10%;">
                            <p style="font-size: 0.6em;">The factorization of the cell type abundance prior $w_{sf}$ and
                                the gene-specific technical sensitivity scaling factor $m_g$ are crucial.</p>
                            <p style="font-size: 0.6em;">When removing fractions of cell types from the reference
                                signatures, the performance of Cell2Location decreases slightly.</p>
                        </div>
                    </div>
                    <aside class="notes">
                        Results of ablation study showing importance of ...
                    </aside>
                </section>

                <section>
                    <h3>Limitations and Potential Improvements</h3>
                    <ul style="font-size: 0.8em;">
                        <li>
                            Limitations: high computational cost and sensitivity to hyperparameter choices.
                        </li>
                        <li>
                            While Cell2Location performs excellently in simulation studies, it competes closely with
                            RCTD in real data analysis.
                        </li>
                        <li>
                            Potential improvements: deep learning accelerates tools (e.g., early stopping, learning rate
                            scheduling, etc.) and applying amortized inference.
                        </li>
                    </ul>
                    <aside class="notes">
                        However, c2l has some limitations, including high computational cost and sensitivity to
                        hyperparameter choices. While c2l performs well in simulation studies, it competes closely with
                        RCTD in real data analysis. Potential improvements include using deep learning to accelerate
                        the training process and applying amortized inference.
                    </aside>
                </section>

                <section>
                    <h4>Accelerating Deep Learning with PyTorch Lightning</h4>
                    <div style=" text-align:center;">
                        <img src="img/pl_quick_start_full_compressed.gif" alt="PyTorch Lightning" style="height: 75%;">
                        <p><small>PyTorch Lightning simplifies deep learning code and accelerates the training
                                process.</small></p>
                    </div>
                    <aside class="notes">
                        Here we introduce pl, a lightweight PyTorch wrapper that simplifies deep learning
                        code and accelerates the training process. It provides a high-level interface for PyTorch and
                        automates the training loop, allowing users to focus on the research rather than the
                        implementation details.
                        Although c2l import pl when importing scvi-tools, it does not use pl to accelerate the
                        training process.
                    </aside>
                </section>
            </section>

            <section>
                <section>
                    <h2>Comparison of RCTD and Cell2Location</h2>
                    <h3>Modeling, Inference, and Simulation</h3>
                    <aside class="notes">
                        Finally, the last section compares RCTD and Cell2Location in terms of modeling, inference, and
                        simulation.
                    </aside>
                </section>

                <section>
                    <h3>Model of RCTD and Cell2Location</h3>
                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 40%; text-align: center;">
                            <p>
                                \[d_{sg}|\lambda_{sg} \sim \text{Poisson}(y_s \lambda_{sg})\]
                            </p>
                            <p style="font-size: 0.8em;">
                                \[
                                \begin{aligned}
                                \log(\lambda_{sg}) &= \alpha_{s} + \log(\sum_f w_{sf} g_{fg})\\
                                &\quad + m_{g} + \epsilon_{sg}
                                \end{aligned}
                                \]
                            </p>
                            <p><small>RCTD Model</small></p>
                        </div>
                        <div style="width: 50%; text-align: center;">
                            <img src="img/c2l_pm.png" alt="Probabilistic Graph" style="width:90%;">
                            <p style="width: 100%; text-align: center;"><small>Probabilistic Graph of
                                    Cell2Location</small></p>
                        </div>
                    </div>
                    <aside class="notes">
                        The basic model is similar because the nb can be view as gamma-poisson mixture. The
                        difference is that c2l has a more complex prior setting.
                    </aside>
                </section>

                <section>
                    <h3>Inference Methods</h3>
                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 50%; text-align: center;">
                            <p>RCTD</p>
                            <p>Sequential Quadratic Programming for MLE</p>
                            <p><small>Convert nonlinear problems into a series of quadratic programming</small></p>
                        </div>
                        <div style="width: 50%; text-align: center;">
                            <p>Cell2Location</p>
                            <p>Automatic Differentiation Variational Inference</p>
                            <p><small>Guide by AutoNormal or Encoder</small></p>
                        </div>
                    </div>
                    <aside class="notes">
                        The inference process of RCTD is generally simpler and faster compared to c2l, enabling quicker
                        results in many analyses.
                    </aside>
                </section>

                <section>
                    <h3>Programming Language and API</h3>
                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 50%; text-align: center;">
                            <p>RCTD: R</p>
                            <p>spacexr</p>
                            <p><small>Cell type-specific differential expression with C-SIDE</small></p>
                        </div>
                        <div style="width: 50%; text-align: center;">
                            <p>Cell2Location: Python</p>
                            <p>scvi-tools / Pyro</p>
                            <p><small>Perform many analysis tasks across single-cell, multi, and spatial omics
                                    data, e.g., dimensionality reduction, FA, auto-annotation, DE, etc.</small></p>
                        </div>
                    </div>
                    <aside class="notes">
                        c2l is build on top of scvi-tools which is a comprehensive toolkit for single-cell data
                        analysis.
                        I believe that the choice of API plays a crucial role in the ease of use and availability of
                        downstream analyses. These
                        two factors are vital for the broader adoption of a method within the research community.
                    </aside>
                </section>

                <section>
                    <h3>Simulation Study</h3>
                    <div style="width: 100%; text-align: center;">
                        <img src="img/cmp_rt.png" alt="Comparison of RCTD and Cell2Location" style="width:80%;">
                    </div>
                    <aside class="notes">
                        From the simulation study, we can see that c2l outperforms RCTD in many scenarios. But c2l is
                        more computationally expensive.
                    </aside>
                </section>

                <section>
                    <h3>Real Data Analysis: DLPFC</h3>
                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 50%; text-align: center;">
                            <p>Human DorsoLateral PreFrontal Cortex (DLPFC)</p>
                            <ul style="font-size: 0.8em;">
                                <li>10 adult neurotypical controls</li>
                                <li>3 positions: anterior, middle, and posterior</li>
                                <li>Annotated snRNA-seq and spatial transcriptomics</li>
                                <li>By 10x Genomics Chromium and Visium</li>
                            </ul>
                        </div>
                        <div style="width: 60%; text-align: center;">
                            <img src="img/DLPFC.jpg" alt="dlpfc" style="width:100%;">
                            <p style="width: 100%; text-align: center; font-size: 0.4em;">Study design to generate
                                paired single nucleus RNA-sequencing (snRNA-seq) and spatially-resolved transcriptomic
                                data across DLPFC.</p>
                        </div>
                    </div>
                    <aside class="notes">
                        We use the DLPFC data from 10X Genomics to compare the performance of RCTD and c2l. The ground
                        truth of st is annotated manually.
                    </aside>
                </section>

                <section>
                    <h4>Real Data Analysis: Data after QC</h4>
                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 65%; text-align: center;">
                            <p>QC: Criteria</p>
                            <ul style="font-size: 0.8em;">
                                <li>Remove cell types with fewer than 25 cells</li>
                                <li>Reduce genes according to the minimum average expression and log fold change</li>
                                <li>Remove genes appearing in fewer than 10 spots</li>
                                <li>Keep genes present in both snRNA-seq and spatial transcriptomics</li>
                                <li>Reduce #gene to around 3k</li>
                            </ul>
                        </div>
                        <div style="width: 35%; text-align: center;">
                            <p>Spatial Transcriptomics</p>
                            <ul style="font-size: 0.8em;">
                                <li>#Spot: 3639</li>
                            </ul>

                            <p>SnRNA-seq</p>
                            <ul style="font-size: 0.8em;">
                                <li>#Gene: about 3k</li>
                                <li>#Cell: 2k-5k</li>
                            </ul>
                        </div>
                    </div>
                    <aside class="notes">
                        We perform qc and result in 3639 spots and 3k genes in snRNA-seq.
                    </aside>
                </section>

                <section>
                    <h4>Real Data Analysis: Layer Level Results</h4>
                    <div style="width: 100%; text-align: center;">
                        <img src="img/res_layer.jpg" alt="result in layer" style="width:100%;">
                        <p style="width: 100%; text-align: center; font-size: 0.6em;">Left panel shows layer level
                            results
                            of two methods in paired DLPFC data. Right panel shows the manual annotation.</p>
                        </p>
                    </div>
                    <aside class="notes">
                        Left panel shows layer level results of two methods in paired DLPFC data. Right panel shows the
                        manual annotation. We can see that c2l provides smoother and less noisy results. But since the
                        paper
                        of c2l suggests to use about 40k epochs and I only use 3k, the result of c2l may not converge
                        which can be seen
                        in the figure that most of the region has small values.
                    </aside>
                </section>

                <section>
                    <h4>Real Data Analysis: Cell Type Level Results</h4>
                    <div style="width: 100%; text-align: center;">
                        <img src="img/res_cell.jpg" alt="result in cell" style="width:100%;">
                        <p style="width: 100%; text-align: center; font-size: 0.6em;">Cell type level results of RCTD
                            and Cell2Location in paired DLPFC data. Manual annotation of two cell types: Astrocytes in
                            layer 1, Micro-Oligodendrocytes in layer 1 as well as white matter. Upper left shows the
                            result of
                            deconvolution visualizing the proportion of a given cell type. Lower left shows the result
                            of mapping assigning spots to the most likely cell type.
                        </p>
                        </p>
                    </div>
                    <aside class="notes">
                        The cell type level results of RCTD and Cell2Location in paired DLPFC data. Manual annotation of
                        two cell types: Astrocytes in layer 1, Micro-Oligodendrocytes in layer 1 as well as white
                        matter.
                        Upper left shows the result of deconvolution visualizing the proportion of a given cell type.
                        Lower left shows the result of mapping assigning spots to the most likely cell type.
                    </aside>
                </section>

                <!-- <section>
                    <h4>Real Data Analysis: Sub Cell Type Level Results</h4>
                    <div style="width: 100%; text-align: center;">
                        <img src="img/res_subcell.jpg" alt="result in cell" style="width:100%;">
                        <p style="width: 100%; text-align: center; font-size: 0.6em;">Sub cell type level results of
                            RCTD and Cell2Location in paired DLPFC data. Manual annotation of Excitatory Neurons 4 in
                            layer 5, while Excitatory Neurons 10 appear in layer 4.
                        </p>
                        </p>
                    </div>
                </section> -->

                <section>
                    <h4>Insights from Real Data Analysis</h4>
                    <div style="font-size: 0.8em;">
                        <ul>
                            <li>The runtime of Cell2Location is significantly longer than RCTD. Deep learning-based
                                methods may not
                                converge if the runtime is limited.</li>
                            <li>Although Cell2Location may not always converge, its results appear to be more accurate
                                than those of
                                RCTD.</li>
                            <li>Compared to RCTD, Cell2Location provides smoother and less noisy results.</li>
                            <li>At the layer level, Cell2Location can leverage spatial information regarding the
                                similarity of spots,
                                which may enhance its performance.</li>
                        </ul>
                    </div>
                    <aside class="notes">
                        .
                    </aside>
                </section>

            </section>

            <section>
                <h3>References</h3>
                <ol style="font-size: 0.6em;">
                    <li>
                        Kleshchevnikov V, Shmatko A, Dann E, et al. Cell2location maps fine-grained cell types in
                        spatial
                        transcriptomics[J]. Nature biotechnology, 2022, 40(5): 661-671. <a
                            href="https://www.nature.com/articles/s41587-021-01139-4">[link]</a>
                    </li>
                    <li>
                        Cable D M, Murray E, Zou L S, et al. Robust decomposition of cell type mixtures in spatial
                        transcriptomics[J]. Nature biotechnology, 2022, 40(4): 517-526. <a
                            href="https://www.nature.com/articles/s41587-021-00830-w">[link]</a>
                    </li>
                    <li>
                        Marx, V. Method of the Year: spatially resolved transcriptomics. Nat Methods 18, 9–14 (2021).
                        <a href="https://doi.org/10.1038/s41592-020-01033-y">[link]</a>
                    </li>
                    <li>Understanding Complexity. Luciano Martelotto. Twitter. 2023.
                        <a href="https://twitter.com/LGMartelotto/status/1186745128615985152">[link]</a>
                    </li>
                    <li>
                        Li B, Zhang W, Guo C, et al. Benchmarking spatial and single-cell transcriptomics integration
                        methods
                        for transcript distribution prediction and cell type deconvolution[J]. Nature methods, 2022,
                        19(6):
                        662-670. <a href="https://www.nature.com/articles/s41592-022-01480-9">[link]</a>
                    </li>
                    <li>
                        Longo S K, Guo M G, Ji A L, et al. Integrating single-cell and spatial transcriptomics to
                        elucidate intercellular tissue
                        dynamics[J]. Nature Reviews Genetics, 2021, 22(10): 627-644.
                        <a href="https://www.nature.com/articles/s41576-021-00370-8">[link]</a>
                    <li>
                        Integrating Single Cell and Visium Spatial Gene Expression Data. 10x Genomics. 2023.
                        <a
                            href="https://www.10xgenomics.com/cn/analysis-guides/integrating-single-cell-and-visium-spatial-gene-expression-data">[link]</a>
                    </li>
                    <li>
                        Spatial Decomposition. Open Problems in Single-Cell Analysis. 2023.
                        <a href="https://openproblems.bio/results/spatial_decomposition/">[link]</a>
                    </li>
                </ol>
            </section>

            <section>
                <p style="font-size: 1.2em; text-align: center;"><strong>References</strong></p>
                <ol style="font-size: 0.6em;" start="9">
                    <li>
                        Huuki-Myers L A, Spangler A, Eagles N J, et al. A data-driven single-cell and spatial
                        transcriptomic
                        map of the human prefrontal cortex[J]. Science, 2024, 384(6698): eadh1938.
                        <a href="https://doi.org/10.1126/science.adh1938">[link]</a>
                    </li>
                    <li>
                        Andersson A, Bergenstråhle J, Asp M, et al. Single-cell and spatial transcriptomics enables
                        probabilistic inference of cell type topography[J]. Communications biology, 2020, 3(1): 565. <a
                            href="https://www.nature.com/articles/s42003-020-01247-y">[link]</a>
                    </li>
                    <li>
                        PyTorch Lightning. 2023.
                        <a href="https://pytorch-lightning.readthedocs.io/en/latest/">[link]</a>
                    </li>
                </ol>
            </section>

        </div>
    </div>

    <script src="dist/reveal.js"></script>
    <script src="plugin/math/math.js"></script>
    <link rel="stylesheet" href="plugin/highlight/github-dark.css" id="highlight-theme" />
    <script src="plugin/highlight/highlight.js"></script>
    <script src="plugin/notes/notes.js"></script>
    <script src="plugin/menu/menu.js"></script>
    <script>
        Reveal.initialize({
            controls: true,
            progress: true,
            center: true,
            hash: true,
            // disableLayout: true,
            transition: 'none',
            slideNumber: true,
            showSlideNumber: 'all',
            help: true,
            mathjax3: {
                mathjax: 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js',
                tex: {
                    inlineMath: [
                        ['$', '$'],
                        ['\\(', '\\)'],
                    ],
                },
                options: {
                    skipHtmlTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
                },
            },
            menu: {
                titleSelector: 'h1, h2, h3',
                hideMissingTitles: true,
            },
            plugins: [RevealMath.MathJax3, RevealHighlight, RevealNotes, RevealMenu],
            // plugins: [RevealMath.MathJax3, RevealHighlight, RevealNotes],
        });
    </script>
</body>

</html>