<!doctype html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <title>Deconvolution of Spatial Transcriptomics Using scRNA-seq</title>

    <meta name="description" content="Deconvolution of Spatial Transcriptomics using scRNA-seq: RCTD and Cell2location">
    <meta name="author" content="Wenxin Jiang">

    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <link rel="stylesheet" href="dist/reset.css">
    <link rel="stylesheet" href="dist/reveal.css">
    <link rel="stylesheet" href="dist/theme/league.css" id="theme">
    <!-- <link rel="stylesheet" href="dist/theme/black.css" id="theme"> -->
    <link rel="stylesheet" href="css/custom.css">
</head>

<body>
    <div class="reveal">
        <div class="slides">

            <section>
                <h2>Deconvolution of Spatial Transcriptomics Using scRNA-seq</h2>
                <p style="text-align: center;">Wenxin Jiang</p>
                <p style="text-align: center;">Oct. 18, 2024</p>
                <aside class="notes">
                    Note write here.
                </aside>
            </section>

            <section>
                <h2>How to Use This Webpage</h2>

                <h3>Click the button at the lower left to open the menu.</h3>

                <h3>Press <span style="color: paleturquoise;">esc</span> on your keyboard to enter navigation mode.</h3>

                <h3>Press <span style="color: royalblue;">?</span> on your keyboard to view keyboard shortcuts.</h3>
            </section>

            <section>
                <h2>Structure of the Presentation</h2>
                <ol>
                    <li>Introduction to <a href="#/3">spatial
                            transcriptomics</a> and <a href="#/4">scRNA-seq</a></li>
                    <li><a href="#/6">Deconvolution methods</a>: <a href="#/7">RCTD</a> and <a
                            href="#/8">Cell2Location</a></li>
                    <li><a href="#/9">Comparison</a>: <a href="#/9/4">Simulation</a> and <a href="#/9/5">Real data</a>
                    </li>
                </ol>
            </section>

            <section>
                <section data-background-image="img/FFPE_Hero_V2_RGB.webp">
                    <h2>Method of the Year: Spatially Resolved Transcriptomics</h2>
                    <!-- ! unknown error in print_pdf mode -->
                    <!-- <div style="position: fixed; bottom: 10px; width: 100%; text-align: center; font-size: 0.6em;">
                        Press the <span style="color: gold;">DOWN</span> arrow to continue
                    </div> -->
                </section>

                <section>
                    <h3>Comparison between Sequencing Methods</h3>
                    <div>
                        <img src="img/spsc_complex.png">
                        <p style="text-align: center;"><small>Understanding the complexity of tissue with bulk RNA-seq,
                                scRNA-seq, and spatial transcriptomics.</small></p>
                    </div>
                </section>

                <section>
                    <h3>Spatial Transcriptomics Maps Gene Expression in Tissue</h3>
                    <div>
                        <img src="img/sp_gene2loc.png">
                        <!-- <p style="text-align: center;">Spatial transcriptomics = Spatial information + gene
                            expression</p> -->
                        <p style="text-align: center;"><small>Spatial transcriptomics combines the spatial information
                                of tissue sections with the gene expression data from RNA sequencing.</small></p>
                    </div>
                </section>

                <section>
                    <!-- TODO: explain seq depth -->
                    <h3>Example: Mouse Hippocampus in 10x Visium</h3>
                    <!-- <table style="width: 100%; table-layout: fixed;">
                        <tr>
                            <td style="width: 60%; text-align: center;">
                                <img src="img/sp_illu.png" alt="Image 1" style="width: 100%; height: auto;">
                                <p><small>Example of spatial transcriptomics</small></p>
                            </td>
                            <td style="width: 60%; text-align: center;">
                                <img src="img/sp_data.png" alt="Image 2" style="width: 40%; height: auto;">
                                <p><small>Structure of the spatial transcriptomics
                                    data.</small></p>
                                </td>
                            </tr>
                        </table> -->
                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 60%; text-align: center;">
                            <img src="img/sp_illu.png" style="width: 100%;">
                            <p><small>Each spot may contain multiple cells or cell types.</small></p>

                        </div>
                        <div style="width: 50%; text-align: center;">
                            <img src="img/sp_data.png" style="width: 45%; margin-left: -10%;">
                            <p style="width: 100%; text-align: center;"><small>Gene expression matrix is highly sparse
                                    count data.</small></p>

                        </div>
                    </div>
                    <p style="text-align: left;"><small>The size of the spots in Visium does not allow us to study gene
                            expression in individual cells. With mixed cell types in spots, we cannot directly map the
                            cell types to the
                            spots.</small></p>
                </section>

                <section>
                    <h3>Deconvolution</h3>
                    <h3>Mapping scRNA-seq to Spatial Transcriptomics</h3>
                    <div>
                        <img src="img/decov_illu.png">
                        <!-- <p style="text-align: center;"><small>Illustration of deconvolution process</small></p> -->
                    </div>
                    <p style="text-align: left;"><small>With the reference scRNA-seq data, we can estimate the cell type
                            composition in each spot of the spatial transcriptomics data.</small></p>
                </section>
            </section>

            <section>
                <section>
                    <h2>Single-Cell RNA Sequencing</h2>
                    <h3>(scRNA-seq)</h3>
                    <div>
                        <img src="img/sc_illu.png" style="width: 70%;">
                        <p style="text-align: center;"><small>ScRNA-seq data provides the gene expression profile of
                                individual cells, which is also sparse count data.</small></p>
                    </div>
                </section>

                <section>
                    <h3>Compare scRNA-seq with Spatial Transcriptomics</h3>
                    <ul>
                        <li><b>Spatial transcriptomics</b>:
                            <ul>
                                <li>Spatial information</li>
                                <li>Typically low cellular resolution</li>
                                <li>Sequence depth is limited</li>
                            </ul>
                        </li>
                        <li><b>ScRNA-seq</b>:
                            <ul>
                                <li>Single-cell resolution</li>
                                <li>High sequence depth</li>
                            </ul>
                        </li>
                    </ul>
                </section>
            </section>

            <section>
                <h2>Insights from Deconvolution Results</h2>
                <div>
                    <img src="img/sp_apply.png" style="width: 70%;">
                </div>
                <p><small>ScRNA-seq and spatial transcriptomics data increase our understanding of the roles of
                        specific cell subpopulations and their interactions in development, homeostasis, and
                        disease.</small></p>
            </section>

            <section>
                <h2>Deconvolution Methods</h2>
                <div>
                    <img src="img/deconv.png" style="width: 70%;">
                    <p style="text-align: center;"><small>Workflow of deconvolution. Up: discovery of cell types in
                            scRNA-seq data. Down: discovery of gene expression in spatial transcriptomics data.</small>
                    </p>
                </div>
            </section>

            <section>
                <section>
                    <h2>RCTD: Robust Cell Type Deconvolution</h2>
                    <div>
                        <img src="img/rctdFig2a.png">
                        <p style="text-align: center;"><small>Illustration of the RCTD method</small></p>
                    </div>
                </section>

                <section>
                    <h3>RCTD Notation of Indexing:</h3>
                    <ul>
                        <li>$s$: index of the spot in spatial transcriptomics dataset</li>
                        <li>$g$: index of the gene</li>
                        <li>$f$: index of the cell type</li>
                    </ul>
                </section>

                <section>
                    <h3>RCTD Notation of Dataset:</h3>
                    <b>Observed data:</b>
                    <ul>
                        <li>$d_{sg}$: observed gene expression count in spot $s$ for gene $g$</li>
                        <li>$y_s$: total transcript count in spot $s$, equals to $\sum_g d_{sg}$</li>
                        <li>$g_{fg}$: mean gene expression for gene $g$ in cell type $f$</li>
                    </ul>
                </section>

                <section>
                    <h3>RCTD Model</h3>
                    <p>
                        \[
                        \begin{aligned}
                        d_{sg}|\lambda_{sg} &\sim \text{Poisson}(y_s \lambda_{sg}) \\
                        \log(\lambda_{sg}) &= \alpha_{s} + \log(\sum_f w_{sf} g_{fg}) + m_{g} + \epsilon_{sg}
                        \end{aligned}
                        \]
                    </p>
                    <p style="text-align: left;">Goal: estimate $w_{sf}$ with $g_{fg}$ estimated from scRNA-seq data.
                    </p>
                </section>

                <section>
                    <h3>RCTD Notation of Dataset:</h3>
                    <b>Unknown parameters:</b>
                    <ul>
                        <li>$w_{sf}$: proportion of cell type $f$ in spot $s$. $\sum_f w_{sf} = 1$ and $w_{sf} \geq 0$
                        </li>
                        <li>$\lambda_{sg}$: parameter of the Poisson distribution for gene $g$ in spot $s$.</li>
                        <li>$\alpha_{s}$: fixed pixel-specific effect for spot $s$</li>
                        <li>$m_{g}$: gene-specific platform random effect for gene $g$</li>
                        <li>$\epsilon_{sg}$: residual error term for gene $g$ in spot $s$</li>
                    </ul>
                </section>

                <section>
                    <h3>RCTD Prior Settings</h3>
                    <ul>
                        <li>$m_{g} \sim \text{Normal}(0, \sigma^2_{m})$</li>
                        <li>$\epsilon_{sg} \sim \text{Normal}(0, \sigma^2_{\epsilon})$.</li>
                    </ul>
                </section>

                <section>
                    <h3>RCTD Model Fitting</h3>
                    <ol>
                        <li>
                            <p><b><span style="color:darkseagreen">Estimate mean gene expression</span></b> $g_{fg}$
                                from scRNA-seq data, denoted as $\hat{g}_{fg}$.
                            </p>
                        </li>
                        <li>
                            <p class="fragment"><b><span style="color:darkseagreen">Gene filtering</span></b> Filter out
                                uninformative genes based on $\hat{g}_{fg}$. Reduce to about 3k genes.</p>
                        </li>
                    </ol>
                </section>

                <section>
                    <h4>RCTD Model Fitting</h4>

                    <!-- <small> -->
                    <div style="font-size: 0.6em; text-align: left;">
                        3. <b><span style="color:darkseagreen">Platform effect normalization</span></b> Estimate $m_{g}$
                        for gene $g$ by summarizing the spatial transcriptomics as a single pseudo-bulk measurement $S_g
                        \equiv \sum_s d_{sg} \sim
                        \text{Poisson}(\bar{d}_{g})$ with
                        \[
                        \begin{aligned}
                        \log(\mathbb{E}[\bar{d}_{g}]|\lambda_{.g}) &= \log\left(\frac{1}{S} \sum_s y_s
                        \lambda_{sg}\right) \\
                        &= m_{g} + \log(\bar{y}\sum_f B_{fg} g_{fg})\\
                        &\approx m_{g} + \log(\bar{y}\sum_f w_{f} g_{fg}) + \log(w_0)
                        \end{aligned}
                        \]
                        where $B_{fg} = \frac{1}{S} \sum_s \frac{y_s}{\bar{y}} w_{sf}
                        \exp(\alpha_{s}+\epsilon_{sg})$
                        and $w_{f} = \frac{1}{S} \sum_s \frac{y_s}{\bar{y}} w_{sf}\alpha_{s}$.

                        Obtain the MLE of unknown parameters ($w_{f}, w_{0}, \sigma^2_{m}$) and solve for $m_{g}$.
                        Denote the predicted platform effect as $\hat{m}_{g}$.
                    </div>
                    <!-- </small> -->

                </section>

                <section>
                    <h4>RCTD Model Fitting</h4>
                    <ol start="4">
                        <li>
                            <p><b><span style="color:darkseagreen">Inference</span></b> Estimate
                                $w_{sf}, \alpha_{s}$ and $\sigma^2_{\epsilon}$ by MLE. Assume estimated $\hat{g}_{fg}$
                                and $\hat{m}_{g}$ are fixed.</p>
                        </li>
                        <li>
                            <p class="fragment"><b><span style="color:darkseagreen">Expected cell-type-specific gene
                                        expression</span></b> $\mathbb{E}[d_{sf}|w, d_{sg}]
                                =\frac{d_{sg}w_{sf}\hat{g}_{fg}}{\sum_{f'} w_{sf'}\hat{g}_{f'g}}$.</p>
                        </li>
                    </ol>
                </section>

                <section>
                    <h3>RCTD Model Selection</h3>
                    <div style="font-size: 0.8em;">
                        <ul>
                            <li>
                                <p><b><span style="color:darkseagreen">Doublet mode</span></b> Assigns
                                    1-2 cell types per spot and is recommended for technologies with high spatial
                                    resolution such as <b>Slide-seq</b> and <b>MERFISH</b>.
                                </p>
                            </li>
                            <li>
                                <p class="fragment"><b><span style="color:darkseagreen">Full mode</span></b> Assigns any
                                    number of cell types per spot and is recommended for technologies with poor spatial
                                    resolution such as <b>100-micron resolution Visium</b>.
                                </p>
                            </li>
                            <li>
                                <p class="fragment"><b><span style="color:darkseagreen">Multi mode</span></b> is an
                                    extension of doublet mode that can discover more than two cell types (up to a
                                    pre-specified amount) per spot as an alternative option to full mode.</p>
                            </li>
                        </ul>
                    </div>
                </section>

                <section>
                    <h3>Cell Type Identification by Model Selection</h3>
                    <p>Denote $\mathcal{L}(f)$ as the likelihood of the model with only $f$ th cell type and
                        $\mathcal{L}(f,
                        f')$ as the likelihood of the model only with $f$th and $f'$th cell types. For each spot $s$:
                    </p>
                    \[
                    \hat{f} = \arg\max_f \mathcal{L}(f) \quad \text{and} \quad \hat{f'} = \arg\max_{f'\neq \hat{f}}
                    \mathcal{L}(\hat{f}, f')
                    \]
                </section>


                <section>
                    <h3>Cell Type Identification by Model Selection</h3>
                    <div style="font-size: 0.8em;">
                        <p style="text-align: left;">Because we expect many pixels to be single cell types, we can apply
                            a <b>penalized approach
                                similar
                                to AIC</b> to decide between the two models. We select the model maximizing:</p>
                        \[
                        \text{AIC}(\mathcal{M}) \equiv \mathcal{L}(\mathcal{M}) - V_p(\mathcal{M})
                        \]
                        <p style="text-align: left;">where $p$ represents the number of parameters (cell types) and $V$
                            represents the penalty
                            weight.
                            Select $V=25$ based on simulation studies.</p>
                    </div>
                </section>

                <section>
                    <h3>Confident and Unconfident Spots</h3>
                    <p>Condition: Existence of another pair of cell types $(f, f')$ (f can be
                        equal to f') such that
                    </p>
                    \[|\mathcal{L}(\hat{f}, \hat{f'}) - \mathcal{L}(f, f')| < \delta\]</p>
                        <p style="text-align: left;">If the condition holds, the spot is <b>unconfident</b>,
                            else it is <b>confident</b>.
                        </p>
                        <!-- <p style="text-align: left;">
                            The threshold $\delta$ is set to 10 based on simulation studies.
                        </p> -->
                </section>

                <section>
                    <h3>Sequential Quadratic Programming for MLE</h3>
                    <p>Aim: Convert nonlinear problems into a series of quadratic programming
                        (QP) problems.</p>
                    <div style="font-size: 0.8em;">
                        <p style="text-align: left;">The parameter $\alpha_s$ effectively allows us to rescale $w_s$,
                            so we define $w_{f, s}=w_{f, s} e^{\alpha_s}$, which <b>will not be constrained to sum to
                                1</b>. Next, define:</p>
                        \[
                        \bar{\lambda}_{s, g}\left(w_s\right)=y_s \sum_{f=1}^F w_{f, s} g_{f, g} e^{\hat{m}_g}=y_s
                        \sum_{f=1}^F
                        w_{f, s} \bar{g}_{f, g}
                        \]
                        <p style="text-align: left;">We will refer to this as the predicted mean of gene $g$ in pixel
                            $s$.</p>
                    </div>
                </section>

                <section>
                    <h4>SQP for MLE (cont.)</h4>
                    <div style="font-size: 0.8em;">
                        <p style="text-align: left;">The final model is a Poisson log-normal mixture model:</p>
                        \[
                        d_{s, g} \mid \bar{\lambda}_{s, g} \sim \operatorname{Poisson}\left(e^{\varepsilon_{s, g}}
                        \bar{\lambda}_{s, g}\left(w_s\right)\right), \quad \varepsilon_{s, g} \sim
                        \operatorname{Normal}\left(0,
                        \sigma_{\varepsilon}^2\right)
                        \]
                        <p style="text-align: left;">We estimate $w_s^* \geq 0$ as the solution that maximizes the
                            <b>log-likelihood</b>
                            $\mathcal{L}\left(w_s\right)$:
                        </p>
                        \[
                        \max \mathcal{L}\left(w_s\right)=\sum_{g=1}^G \log P\left(d_{s, g} \mid \bar{\lambda}_{s,
                        g}\left(w_s\right)\right) \quad \text { subject to: } w \geq 0
                        \]
                    </div>
                </section>

                <section>
                    <h4>SQP for MLE (cont.)</h4>
                    <div style="font-size: 0.8em;">
                        <p style="text-align: left;">From now on, we consider a fixed spot $s$ and suppress the notation
                            of $s$. We will estimate
                            $w^*
                            \geq 0$ as that which maximizes the <b>log-likelihood</b> $\mathcal{L}(w)$:</p>
                        \[
                        \max _w \mathcal{L}(w)=\sum_{g=1}^G \log P\left(d_g \mid \lambda_g(w)\right)=\sum_{g=1}^G \log
                        Q_{d_g}\left(\lambda_g(w)\right)
                        \]
                    </div>
                </section>

                <section>
                    <h4>SQP for MLE (cont.)</h4>
                    <div style="font-size: 0.8em;">
                        <p style="text-align: left;">Our log likelihood is non-convex. To optimize it, we will apply
                            Sequential Quadratic
                            Programming,
                            an
                            iterative procedure that will be repeated until convergence. Let $w_0$ be the value of $w$
                            at a
                            given iteration, and let the gradient of $-\mathcal{L}$ be $b(w)$ and the Hessian of
                            $-\mathcal{L}$
                            be $A(w)$. Then, we can make the following quadratic Taylor approximation to $\mathcal{L}$:
                        </p>
                        \[
                        \begin{aligned}
                        -\mathcal{L}(w)
                        \approx&-\mathcal{L}\left(w_0\right)+b\left(w_0\right)^T\left(w-w_0\right)\\
                        &+\frac{1}{2}\left(w-w_0\right)^T A\left(w_0\right)\left(w-w_0\right)
                        \end{aligned}
                        \]
                    </div>
                </section>

                <section>
                    <h3>SQP for MLE: Construct Positive Semi-Definite Hessian</h3>
                    <p style="font-size: 0.8em; text-align: left;">This QP will not be well-behaved if the Hessian
                        $A\left(w_0\right)$ is not positive semi-definite, which can occur due to the non-convexity of
                        $-\mathcal{L}(w)$. Specifically, suppose we have an eigen-decomposition of $H$ as:</p>
                    \[
                    H=V D V^T
                    \]
                    <p style="font-size: 0.8em; text-align: left;">Here, $D$ is a diagonal matrix of eigenvalues. We
                        obtain the positive semi-definite part of H by taking $D^{+}=\max (D, 0)$ and:</p>
                    \[
                    A=V D^{+} V^T
                    \]
                </section>

                <section>
                    <h3>Advantages of RCTD</h3>
                    <ul>
                        <li>Models platform effects by taking spatial transcriptomics as a pseudo-bulk
                            measurement.
                        </li>
                        <li>Robust to varying UMI counts per pixel, multiple cell types per pixel, and missing cell
                            types
                            in
                            the reference.</li>
                    </ul>
                </section>

                <section>
                    <h3>Platform Effect Estimation</h3>
                    <div style="text-align:center;">
                        <img src="img/rctdFig1f.png" alt="Figure 1f" style="width:40%;">
                        <p><small>Density plot across genes of measured platform effects between cerebellum
                                <b>scRNA-seq</b> and <b>snRNA-seq</b> data.</small></p>
                    </div>
                </section>

                <section>
                    <h3>Platform Effect Estimation</h3>
                    <div style="text-align:center; font-size: 0.6em;">
                        <p>In the 3rd step of model fitting, RCTD estimates the platform effect of each gene to transfer
                            cell type information from scRNA-seq to spatial transcriptomics by taking the spatial
                            transcriptomics as a bulk measurement.
                        </p>
                    </div>

                    <div style="text-align:center;">
                        <img src="img/rctdFig2bc.png" alt="Figure 2bc" style="width:60%;">
                        <p><small>Left: Scatter plot of measured versus predicted platform effect for each gene between
                                the sc and sn cerebellum datasets. Right: Confusion matrix for RCTD’s performance on
                                cross-platform (trained on snRNA-seq data and tested on scRNA-seq data).</small></p>
                    </div>
                </section>

                <section>
                    <h3>Robustness</h3>
                    <div style="font-size: 0.8em;">
                        <ul>
                            <li>
                                <p><b><span style="color:darkseagreen">Varying unique molecular
                                            identifier (UMI) counts per pixel:</b></span> Additional UMIs per
                                    pixel lead to an increased confidence rate.</p>
                            </li>
                            <li>
                                <p class="fragment"><b><span style="color:darkseagreen">Multiple cell types per
                                            pixel:</b></span> RCTD was able to accurately predict cell class
                                    proportions on
                                    pixels containing three or four cell types.</p>
                            </li>
                            <li>
                                <p class="fragment"><b><span style="color:darkseagreen">Missing cell types in the
                                            reference:</b></span>When cell types in the simulated spatial data were
                                    missing
                                    from the reference, RCTD classified pixels as the most transcriptionally similar
                                    cell
                                    type in the reference if available. When no closest cell type was available in the
                                    reference, RCTD predicted cell types with reduced confidence rates but often
                                    misclassified such pixels.</p>
                            </li>
                        </ul>
                    </div>
                </section>

                <section>
                    <h3>Application: Detecting Spatially Variable Genes and Effect of Cellular Environment</h3>
                    <p style="text-align: left; font-size: 0.8em;">Differentiate cell type marker genes from spatially
                        variable genes.</p>

                    <div style="text-align:center;">
                        <img src="img/rctdFig6d.png" alt="Figure 6d" style="width:70%;">
                        <p><small>Smoothed spatial expression patterns, recovered by local regression, of two genes
                                detected to have large spatial variation within RCTD’s CA3 cells.</small></p>
                    </div>
                </section>

                <section>
                    <h3>Limitations</h3>
                    <ol>
                        <li>Assumes that platform effects are shared across cell types and random noise $\epsilon_{fg}$
                            is
                            shared across genes.</li>
                        <li>Hard to handle cases when the cell type is missing from the reference but exists in the
                            spatial
                            data.</li>
                    </ol>
                </section>
            </section>

            <section>
                <section>
                    <h2>Cell2Location</h2>
                    <div>
                        <img src="img/c2lFig1a.png">
                        <p style="text-align: center;"><small>Illustration of the Cell2Location method</small></p>
                    </div>
                </section>

                <section>
                    <h3>Cell2Location Notation of Index</h3>
                    <ul>
                        <li>$g \in \{1, \ldots, G\}$: Gene</li>
                        <li>$s \in \{1, \ldots, S\}$: Spot</li>
                        <li>$e \in \{1, \ldots, E\}$: Spatial dataset</li>
                        <li>$f \in \{1, \ldots, F\}$: Cell type</li>
                    </ul>
                </section>

                <section>
                    <h3>Cell2Location Notation of Dataset</h3>
                    <p><b>Input</b>: Matrix of reference expression levels $g_{fg}$ and matrix of spatial expression
                        counts
                        $d_{sg}$.</p>
                    <p><b>Main output</b>: Regression weights $w_{sf}$ which can be transformed into cell type
                        proportions.
                    </p>
                </section>

                <section>
                    <h3>Cell2Location Model of mRNA Counts</h3>
                    <!-- <div style="display: flex; justify-content: space-between; align-items: flex-start;">
                        <div style="width: 38%; text-align: left;">
                            <div style="font-size: 0.8em;">
                                \[
                                \begin{aligned}
                                d_{sg} &\sim \text{NB}(\mu_{sg}, \alpha_{eg}) \\
                                \mu_{sg} &= \left( m_{g} \sum_{f} w_{sf} g_{fg} + s_{eg} \right) y_s
                                \end{aligned}
                                \]
                            </div>
                        </div>
                        <div style="width: 60%; text-align: right;">
                            <img src="img/c2lSFig1a.png" alt="SFigure 1f" style="width: 90%;">
                        </div>
                    </div> -->
                    <div style="font-size: 0.8em;">
                        \[
                        d_{sg} \sim \text{NB}(\mu_{sg}, \alpha_{eg}), \quad
                        \mu_{sg} = \left( m_{g} \sum_{f} w_{sf} g_{fg} + s_{eg} \right) y_s
                        \]
                    </div>
                    <div style="text-align: center;">
                        <img src="img/c2lSFig1a.png" alt="SFigure 1f" style="width: 90%;">
                    </div>
                </section>

                <section>
                    <p><b>Cell2Location Notation of Model Parameters</b></p>
                    <div style="font-size: 0.9em;">
                        <ul>
                            <li>$d_{sg}$: Spatial expression of gene $g$ in spot $s$</li>
                            <li>$\mu_{sg}$: Unobserved expression level (rate) of gene $g$ in spot $s$</li>
                            <li>$\alpha_{eg}$: Gene- and batch-specific over-dispersion parameter</li>
                            <li>$m_{g}$: Technology sensitivity parameter for gene $g$</li>
                            <li>$w_{sf}$: Regression weight for cell type $f$ in spot $s$</li>
                            <li>$g_{fg}$: Reference expression level of gene $g$ in cell type $f$</li>
                            <li>$s_{eg}$: Gene-specific additive shift in spatial dataset $e$</li>
                            <li>$y_s$: Location-specific scaling factor for spot $s$</li>
                        </ul>
                    </div>
                </section>

                <section>
                    <h3>Cell2Location Cell Abundance Prior</h3>
                    <img src="img/c2lSFig1b.png" alt="SFigure 1f" style="width:90%;">
                </section>

                <section>
                    <h3>Prior Settings of Cell2Location</h3>
                    <p style="font-size: 0.8em; text-align: left;">Gene-specific multiplicative scaling factor</p>
                    <div style="font-size: 0.6em;">
                        \[
                        \begin{aligned}
                        m_g &\sim \text{Gamma}\left(\alpha^m, \alpha^m/\mu^m \right)
                        \left\{ \begin{array}{ll}
                        \alpha^m & = 1/(o^m)^2, \quad o^m \sim \text{Exp}(3) \\
                        \mu^m &\sim \text{Gamma}(1, 1) \\
                        \end{array} \right. \\
                        \end{aligned}
                        \]
                    </div>

                    <p style="font-size: 0.8em; text-align: left;">The prior on detection efficiency per location</p>
                    <div style="font-size: 0.6em;">
                        \[
                        \begin{aligned}
                        y_s &\sim \text{Gamma}\left(\alpha^y, \alpha^y/y_e \right) \quad y_e \sim \text{Gamma}(10,
                        10/\mu^y)
                        \end{aligned}
                        \]
                    </div>

                    <p style="font-size: 0.8em; text-align: left;">Overdispersion for each gene</p>
                    <div style="font-size: 0.6em;">
                        \[
                        \alpha_{eg} = 1/o_{eg}^2, \quad o_{eg} \sim \text{Exp}(\beta^o), \quad \beta^o \sim
                        \text{Gamma}(9,
                        3)
                        \]
                    </div>
                </section>

                <section>
                    <p style="font-size: 0.8em; text-align: left;">Additive shift for genes</p>
                    <div style="font-size: 0.6em;">
                        \[
                        \begin{aligned}
                        s_{eg} &\sim \text{Gamma}\left(\alpha^s_e, \alpha^s_e/\mu^s_e \right)
                        \left\{ \begin{array}{ll}
                        \mu^s_e &\sim \text{Gamma}(1, 100) \\
                        \alpha^s_e &=1/o_e^2, \quad o_e \sim \text{Exp}(\beta^s), \quad \beta^s \sim \text{Gamma}(9, 3)
                        \end{array} \right.
                        \end{aligned}
                        \]
                    </div>

                    <p style="font-size: 0.8em; text-align: left;">Cell abundance prior:</p>
                    <div style="font-size: 0.6em;">
                        \[
                        \begin{aligned}
                        w_{sf} &\sim \text{Gamma}\left(\sum_r z_{sr} x_{rf} \nu^w, \nu^w\right) \\
                        z_{sr} &\sim \text{Gamma}\left(B_s/R, 1/(N_s/B_s)\right)
                        \left\{ \begin{array}{ll}
                        N_s &\sim \text{Gamma}\left(\hat{N} \nu^n, \nu^n\right) \\
                        B_s &\sim \text{Gamma}\left(\hat{B}, 1\right) \\
                        \end{array} \right. \\
                        x_{rf} &\sim \text{Gamma}\left(K_r/R, K_r\right), \quad K_r \sim
                        \text{Gamma}\left(\hat{A}/\hat{B},
                        1\right) \\
                        \end{aligned}
                        \]
                    </div>
                </section>

                <section>
                    <h3>Cell2Location Model: Probabilistic Graph</h3>
                    <div style="text-align:center;">
                        <img src="img/c2l_pm.png" alt="Probabilistic Graph" style="width:85%;">
                        <!-- <p><small></small></p> -->

                    </div>
                </section>

                <section>
                    <h3>Strength of the Cell2Location Model</h3>
                    <p style="font-size: 0.8em; text-align: left;">Joint modeling of multiple spatial
                        experiments/batches provides several benefits due to normalization and sharing of information
                        between experiments:</p>

                    <div style="font-size: 0.8em;">
                        <ul>
                            <li>
                                <p class="fragment">Modeling differences in RNA detection sensitivity across
                                    experiments: $y_e$ and $y_s$.</p>
                            </li>
                            <li>
                                <p class="fragment">Increasing accuracy by improving the model's ability to
                                    distinguish low sensitivity
                                    $m_g$ from zero cell abundance $w_{rf}$.</p>
                            </li>
                            <li>
                                <p class="fragment">Increasing sensitivity by sharing factorization prior on cell
                                    abundance $w_{rf}$, namely
                                    which cell types tend to co-locate across experiments represented by $x_{rf}$.</p>
                            </li>
                        </ul>
                    </div>
                </section>

                <section>
                    <h3>Fitting the Cell2Location Model</h3>
                    <h4>Step 1: Construction of Reference Cell Type Signatures</h4>

                    <div style="font-size: 0.8em;">
                        <ul>
                            <li>
                                <p class="fragment">
                                    <scan style="color:darkseagreen">Distribution</scan>: By default,
                                    the Cell2Location employs a <b>negative binomial regression</b> to estimate
                                    reference
                                    cell type signatures, which allows robust combination of data from multiple sources.
                                </p>
                            </li>
                            <li>
                                <p class="fragment">
                                    <scan style="color:darkseagreen">Mean</scan>: Alternatively, a
                                    <b>hand-coded method</b> that estimates the average expression of each gene in each
                                    cell type can be used.
                                </p>
                            </li>
                        </ul>
                    </div>
                </section>

                <section>
                    <h4>Fitting the Cell2Location Model</h4>
                    <h4>Step 2: Determining the Hyperparameters</h4>
                    <div style="font-size: 0.8em;">
                        <ul>
                            <li>
                                <p class="fragment">
                                    <scan style="color:darkseagreen">Expected cell abundance $\hat{N}$ per location
                                    </scan>: Can be derived from histology images or tissue type.
                                </p>
                            </li>
                            <li>
                                <p class="fragment">
                                    <scan style="color:darkseagreen">Hyperparameter $\alpha^y$ for regularizing
                                        within-experiment variation in RNA detection sensitivity</scan>: If not strong
                                    within-experiment variability in RNA detection sensitivity across locations, set
                                    $\alpha^y=200$. Otherwise, set $\alpha^y=20$.
                                </p>
                            </li>
                        </ul>
                    </div>
                </section>

                <section>
                    <h4>Fitting the Cell2Location Model</h4>
                    <h4>Step 3: Variational Inference</h4>
                    <p style="text-align: left; font-size: 0.8em;">Variational Bayesian Inference is used to approximate
                        the posterior, building on the Automatic Differentiation Variational Inference (<scan
                            style="color:darkseagreen">ADVI</scan>) framework
                        implemented in Pyro.</p>
                    <p style="text-align: left; font-size: 0.6em;">The posterior distribution over unknown parameters
                        is approximated by softplus-transformed (to ensure a positive scale) <scan
                            style="color:darkseagreen">univariate normal</scan> distributions.</p>
                    <p style="text-align: left; font-size: 0.6em;">
                        <scan style="color:darkseagreen">Minimizing the KL divergence</scan> between the variational
                        approximation and the true posterior
                        distribution (or maximizing ELBO).
                    </p>
                </section>

                <section>
                    <h3>Brief Introduction to Pyro</h3>
                    <p style="text-align: left; font-size: 0.8em;">Here's an example to illustrate the use of Pyro for
                        inference:</p>

                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 73%; text-align: center;">
                            <div style="font-size: 0.7em;">
                                <pre><code data-line-numbers="5,6,8,11,13-16" class="hljs python"">data = np.random.binomial(1, 0.8, size=200) # random-generated data
                    
def model(data): # define the probabilistic model
    # define the hyperparameters that control the Beta prior
    alpha0 = pyro.param("alpha0", torch.tensor(10.0))
    beta0 = pyro.param("beta0", torch.tensor(10.0))
    # sample f from the Beta prior
    f = pyro.sample("latent_var", dist.Beta(alpha0, beta0))
    # sample observed data from the Bernoulli likelihood
    with pyro.plate("data", len(data)):
        pyro.sample("obs", dist.Bernoulli(f), obs=data)

svi = SVI(model,
            AutoDiagonalNormal(model),
            Adam({"lr": 0.0005, "betas": (0.90, 0.999)}),
            loss=Trace_ELBO())

[svi.step(data) for _ in range(1000)] # run inference
                                </code></pre>
                            </div>
                        </div>
                        <div style="width: 30%; text-align: right; margin-left: -8em;">
                            <img src="img/pyro_structure.svg" alt="Pyro structure" style="width: 100%;">
                        </div>
                    </div>
                </section>

                <section>
                    <h3>Features of Cell2Location</h3>
                    <div style="font-size: 0.8em;">
                        <ul>
                            <li>
                                <p><b><span style="color:darkseagreen">Borrow statistical strength
                                            across locations</span> with similar cell composition</b></p>
                            </li>
                            <li>
                                <p class="fragment">Account for <b><span style="color:darkseagreen">batch
                                            variation</span>
                                        across slides as well as variation in <span style="color: darkseagreen;">mRNA
                                            detection sensitivity<span></b>
                                </p>
                            </li>
                            <li>
                                <p class="fragment"><b>Estimate absolute cell type abundances by incorporating <span
                                            style="color:darkseagreen">prior information</span> about the analyzed
                                        tissues</b>
                                </p>
                            </li>
                            <li>
                                <p class="fragment"><b><span style="color:darkseagreen">Computationally
                                            efficient</span>,
                                        owing to variational approximate inference and GPU acceleration</b>
                                </p>
                            </li>
                            <li>
                                <p class="fragment"><b>Integrated with the
                                        <span style="color:darkseagreen">scvi-tools</span> framework and comes with a
                                        suite of downstream analysis
                                        tools</b></p>
                            </li>
                        </ul>
                    </div>
                </section>

                <section>
                    <h3>Sensitivity Analysis of Cell2Location</h3>
                    <p style="text-align: left; font-size: 0.8em;">Cell2Location is sensitive to the choice of
                        $\alpha^y$ and $\hat{N}$ but robust to $\hat{A}$ and
                        $\hat{B}$.</p>
                    <div style="text-align:center;">
                        <img src="img/c2lsena.png" alt="SFigure 5" style="width:90%;">
                        <!-- <img src="img/c2lsenb.png" alt="SFigure 5" style="width:90%;"> -->
                        <!-- <p><small><b>SupFigure 5</b>: Stability of cell2location cell abundance estimates with respect
                                to
                                model hyperparameter settings.</small></p> -->
                    </div>
                </section>

                <section>
                    <h4>Ablation Study and Unaligned Cell Type Signatures</h4>
                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 70%; text-align: center;">
                            <img src="img/c2lablation.png" alt="SFigure 4" style="width:100%;">
                        </div>

                        <div style="width: 30%; text-align: left; margin-left:-10%;">
                            <p style="font-size: 0.6em;">The factorization of the cell type abundance prior $w_{sf}$ and
                                the gene-specific technical sensitivity scaling factor $m_g$ are crucial.</p>
                            <p style="font-size: 0.6em;">When removing fractions of cell types from the reference
                                signatures, the performance of Cell2Location decreases slightly.</p>
                        </div>
                    </div>
                </section>

                <section>
                    <h3>Limitations and Potential Improvements</h3>
                    <ul style="font-size: 0.8em;">
                        <li>
                            Limitations: high computational cost and sensitivity to hyperparameter choices.
                        </li>
                        <li>
                            While Cell2Location performs excellently in simulation studies, it competes closely with
                            RCTD in real data analysis.
                        </li>
                        <li>
                            Potential improvements: deep learning accelerates tools (e.g., early stopping, learning rate
                            scheduling, etc.) and applying amortized inference.
                        </li>
                    </ul>
                </section>

                <section>
                    <h4>Accelerating Deep Learning with PyTorch Lightning</h4>
                    <div style=" text-align:center;">
                        <img src="img/pl_quick_start_full_compressed.gif" alt="PyTorch Lightning" style="height: 80%;">
                        <p><small>PyTorch Lightning simplifies deep learning code and accelerates the training
                                process.</small></p>
                    </div>
                </section>
            </section>



            <section>
                <section>
                    <h2>Comparison of RCTD and Cell2Location</h2>
                    <h3>Modeling, Inference, and Simulation</h3>
                </section>

                <section>
                    <h3>Model of RCTD and Cell2Location</h3>
                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 40%; text-align: center;">
                            <p>
                                \[d_{sg}|\lambda_{sg} \sim \text{Poisson}(y_s \lambda_{sg})\]
                            </p>
                            <p style="font-size: 0.8em;">
                                \[
                                \begin{aligned}
                                \log(\lambda_{sg}) &= \alpha_{s} + \log(\sum_f w_{sf} g_{fg})\\
                                &\quad + m_{g} + \epsilon_{sg}
                                \end{aligned}
                                \]
                            </p>
                            <p><small>RCTD Model</small></p>
                        </div>
                        <div style="width: 50%; text-align: center;">
                            <img src="img/c2l_pm.png" alt="Probabilistic Graph" style="width:90%;">
                            <p style="width: 100%; text-align: center;"><small>Probabilistic Graph of
                                    Cell2Location</small></p>
                        </div>
                    </div>
                </section>

                <section>
                    <h3>Inference Methods</h3>
                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 50%; text-align: center;">
                            <p>RCTD</p>
                            <p>Sequential Quadratic Programming for MLE</p>
                            <p><small>Convert nonlinear problems into a series of quadratic programming</small></p>
                        </div>
                        <div style="width: 50%; text-align: center;">
                            <p>Cell2Location</p>
                            <p>Automatic Differentiation Variational Inference</p>
                            <p><small>Guide by AutoNormal or Encoder</small></p>
                        </div>
                    </div>
                </section>

                <section>
                    <h3>Programming Language and API</h3>
                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 50%; text-align: center;">
                            <p>RCTD: R</p>
                            <p>spacexr</p>
                            <p><small>Cell type-specific differential expression with C-SIDE</small></p>
                        </div>
                        <div style="width: 50%; text-align: center;">
                            <p>Cell2Location: Python</p>
                            <p>scvi-tools / Pyro</p>
                            <p><small>Perform many analysis tasks across single-cell, multi, and spatial omics
                                    data, e.g., dimensionality reduction, FA, auto-annotation, DE, etc.</small></p>
                        </div>
                    </div>
                </section>

                <section>
                    <h3>Simulation Study</h3>
                    <div style="width: 100%; text-align: center;">
                        <img src="img/cmp_rt.png" alt="Comparison of RCTD and Cell2Location" style="width:80%;">
                    </div>
                </section>

                <section>
                    <h3>Real Data Analysis: DLPFC</h3>
                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 50%; text-align: center;">
                            <p>Human DorsoLateral PreFrontal Cortex (DLPFC)</p>
                            <ul style="font-size: 0.8em;">
                                <li>10 adult neurotypical controls</li>
                                <li>3 positions: anterior, middle, and posterior</li>
                                <li>Annotated snRNA-seq and spatial transcriptomics</li>
                                <li>By 10x Genomics Chromium and Visium</li>
                            </ul>
                        </div>
                        <div style="width: 60%; text-align: center;">
                            <img src="img/DLPFC.jpg" alt="dlpfc" style="width:100%;">
                            <p style="width: 100%; text-align: center; font-size: 0.4em;">Study design to generate
                                paired single nucleus RNA-sequencing (snRNA-seq) and spatially-resolved transcriptomic
                                data across DLPFC.</p>
                        </div>
                    </div>
                </section>

                <section>
                    <h4>Real Data Analysis: Data after QC</h4>
                    <div style="width: 100%; display: flex; align-items: center; justify-content: space-between;">
                        <div style="width: 65%; text-align: center;">
                            <p>QC: Criteria</p>
                            <ul style="font-size: 0.8em;">
                                <li>Remove cell types with fewer than 25 cells</li>
                                <li>Reduce genes according to the minimum average expression and log fold change</li>
                                <li>Remove genes appearing in fewer than 10 spots</li>
                                <li>Keep genes present in both snRNA-seq and spatial transcriptomics</li>
                                <li>Reduce #gene to around 3k</li>
                            </ul>
                        </div>
                        <div style="width: 35%; text-align: center;">
                            <p>Spatial Transcriptomics</p>
                            <ul style="font-size: 0.8em;">
                                <li>#Spot: 3639</li>
                            </ul>

                            <p>SnRNA-seq</p>
                            <ul style="font-size: 0.8em;">
                                <li>#Gene: about 3k</li>
                                <li>#Cell: 2k-5k</li>
                            </ul>
                        </div>
                    </div>
                </section>

                <section>
                    <h4>Real Data Analysis: Layer Level Results</h4>
                    <div style="width: 100%; text-align: center;">
                        <img src="img/res_layer.jpg" alt="result in layer" style="width:100%;">
                        <p style="width: 100%; text-align: center; font-size: 0.6em;">Left panel shows layer level
                            results
                            of two methods in paired DLPFC data. Right panel shows the manual annotation.</p>
                        </p>
                    </div>
                </section>

                <section>
                    <h4>Real Data Analysis: Cell Type Level Results</h4>
                    <div style="width: 100%; text-align: center;">
                        <img src="img/res_cell.jpg" alt="result in cell" style="width:100%;">
                        <p style="width: 100%; text-align: center; font-size: 0.6em;">Cell type level results of RCTD
                            and Cell2Location in paired DLPFC data. Manual annotation of two cell types: Astrocytes in
                            layer 1, Micro-Oligodendrocytes in layer 1 as well as white matter. Upper left shows the
                            result of
                            deconvolution visualizing the proportion of a given cell type. Lower left shows the result
                            of mapping assigning spots to the most likely cell type.
                        </p>
                        </p>
                    </div>
                </section>

                <!-- <section>
                    <h4>Real Data Analysis: Sub Cell Type Level Results</h4>
                    <div style="width: 100%; text-align: center;">
                        <img src="img/res_subcell.jpg" alt="result in cell" style="width:100%;">
                        <p style="width: 100%; text-align: center; font-size: 0.6em;">Sub cell type level results of
                            RCTD and Cell2Location in paired DLPFC data. Manual annotation of Excitatory Neurons 4 in
                            layer 5, while Excitatory Neurons 10 appear in layer 4.
                        </p>
                        </p>
                    </div>
                </section> -->

                <section>
                    <h4>Insights from Real Data Analysis</h4>
                    <div style="font-size: 0.8em;">
                        <ul>
                            <li>The runtime of Cell2Location is significantly longer than RCTD. Deep learning-based
                                methods may not
                                converge if the runtime is limited.</li>
                            <li>Although Cell2Location may not always converge, its results appear to be more accurate
                                than those of
                                RCTD.</li>
                            <li>Compared to RCTD, Cell2Location provides smoother and less noisy results.</li>
                            <li>At the layer level, Cell2Location can leverage spatial information regarding the
                                similarity of spots,
                                which may enhance its performance.</li>
                        </ul>
                    </div>
                </section>

            </section>

            <section>
                <h3>References</h3>
                <ol style="font-size: 0.6em;">
                    <li>
                        Kleshchevnikov V, Shmatko A, Dann E, et al. Cell2location maps fine-grained cell types in
                        spatial
                        transcriptomics[J]. Nature biotechnology, 2022, 40(5): 661-671. <a
                            href="https://www.nature.com/articles/s41587-021-01139-4">[link]</a>
                    </li>
                    <li>
                        Cable D M, Murray E, Zou L S, et al. Robust decomposition of cell type mixtures in spatial
                        transcriptomics[J]. Nature biotechnology, 2022, 40(4): 517-526. <a
                            href="https://www.nature.com/articles/s41587-021-00830-w">[link]</a>
                    </li>
                    <li>
                        Marx, V. Method of the Year: spatially resolved transcriptomics. Nat Methods 18, 9–14 (2021).
                        <a href="https://doi.org/10.1038/s41592-020-01033-y">[link]</a>
                    </li>
                    <li>
                        Li B, Zhang W, Guo C, et al. Benchmarking spatial and single-cell transcriptomics integration
                        methods
                        for transcript distribution prediction and cell type deconvolution[J]. Nature methods, 2022,
                        19(6):
                        662-670.
                    </li>
                    <li>
                        Integrating Single Cell and Visium Spatial Gene Expression Data. 10x Genomics. 2023.
                        <a
                            href="https://www.10xgenomics.com/cn/analysis-guides/integrating-single-cell-and-visium-spatial-gene-expression-data">[link]</a>
                    </li>
                    <li>
                        Spatial Decomposition. Open Problems in Single-Cell Analysis. 2023.
                        <a href="https://openproblems.bio/results/spatial_decomposition/">[link]</a>
                    </li>
                    <li>
                        Huuki-Myers L A, Spangler A, Eagles N J, et al. A data-driven single-cell and spatial
                        transcriptomic
                        map of the human prefrontal cortex[J]. Science, 2024, 384(6698): eadh1938.
                        <a href="https://doi.org/10.1126/science.adh1938">[link]</a>
                    </li>
                    <li>
                        PyTorch Lightning. 2023.
                        <a href="https://pytorch-lightning.readthedocs.io/en/latest/">[link]</a>
                    </li>
                </ol>
            </section>

        </div>
    </div>

    <script src="dist/reveal.js"></script>
    <script src="plugin/math/math.js"></script>
    <link rel="stylesheet" href="plugin/highlight/github-dark.css" id="highlight-theme" />
    <script src="plugin/highlight/highlight.js"></script>
    <script src="plugin/notes/notes.js"></script>
    <script src="plugin/menu/menu.js"></script>
    <script>
        Reveal.initialize({
            controls: true,
            progress: true,
            center: true,
            hash: true,
            // disableLayout: true,
            transition: 'none',
            slideNumber: true,
            showSlideNumber: 'all',
            help: true,
            mathjax3: {
                mathjax: 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js',
                tex: {
                    inlineMath: [
                        ['$', '$'],
                        ['\\(', '\\)'],
                    ],
                },
                options: {
                    skipHtmlTags: ['script', 'noscript', 'style', 'textarea', 'pre'],
                },
            },
            menu: {
                titleSelector: 'h1, h2, h3',
                hideMissingTitles: true,
            },
            plugins: [RevealMath.MathJax3, RevealHighlight, RevealNotes, RevealMenu],
            // plugins: [RevealMath.MathJax3, RevealHighlight, RevealNotes],
        });
    </script>
</body>

</html>