# Presentation Materials for the "Deconvolution of Spatial Transcriptomics Using scRNA-seq"

## Introduction

Spatial transcriptomics is a powerful tool for studying the spatial organization of cells in tissues. It provides a high-resolution map of gene expression in the context of tissue architecture. However, the interpretation of spatial transcriptomics data can be challenging due to the complexity of the data and the presence of multiple cell types in the tissue. Deconvolution methods aim to infer the spatial distribution of cell types in the tissue based on spatial transcriptomics data and single-cell RNA sequencing (scRNA-seq) data.

In this presentation, we will discuss the challenges of deconvolution of spatial transcriptomics data using scRNA-seq data and review recent advances in this field. We will also provide an overview of the RCTD and Cell2Location methods and discuss their strengths and limitations.

Your can visit [https://lucajiang.github.io/ST_Deconv/](https://lucajiang.github.io/ST_Deconv/) to view the latest version of presentation slides.

You can also download the PDF version of the slides from [https://lucajiang.github.io/ST_Deconv/?print-pdf](https://lucajiang.github.io/ST_Deconv/?print-pdf).
