# Reference
1. Kleshchevnikov V, Shmatko A, Dann E, et al. Cell2location maps fine-grained cell types in spatial transcriptomics[J]. Nature biotechnology, 2022, 40(5): 661-671.[link](https://www.nature.com/articles/s41587-021-01139-4)
2. Cable D M, Murray E, Zou L S, et al. Robust decomposition of cell type mixtures in spatial transcriptomics[J]. Nature biotechnology, 2022, 40(4): 517-526.[link](https://www.nature.com/articles/s41587-021-00830-w)
2. Marx, V. Method of the Year: spatially resolved transcriptomics. Nat Methods 18, 9â€“14 (2021). [link]([https://](https://doi.org/10.1038/s41592-020-01033-y))
3. Li B, Zhang W, Guo C, et al. Benchmarking spatial and single-cell transcriptomics integration methods for transcript distribution prediction and cell type deconvolution[J]. Nature methods, 2022, 19(6): 662-670.
4. Integrating Single Cell and Visium Spatial Gene Expression Data. 10x Genomics. 2023.[link](https://www.10xgenomics.com/cn/analysis-guides/integrating-single-cell-and-visium-spatial-gene-expression-data)
5. Spatial Decomposition. Open Problems in Single-Cell Analysis. 2023. [link](https://openproblems.bio/results/spatial_decomposition/)
6. Huuki-Myers L A, Spangler A, Eagles N J, et al. A data-driven single-cell and spatial transcriptomic map of the human prefrontal cortex[J]. Science, 2024, 384(6698): eadh1938. [link](https://doi.org/10.1126/science.adh1938)
